{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import os\n",
    "import re"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merged CREs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Gene_Mapping/refs/unique_cres.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>797406.0</td>\n",
       "      <td>798178.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>816037.0</td>\n",
       "      <td>818244.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>819296.0</td>\n",
       "      <td>820336.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>826435.0</td>\n",
       "      <td>828307.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>842686.0</td>\n",
       "      <td>843233.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200044</th>\n",
       "      <td>200045</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21421806.0</td>\n",
       "      <td>21422749.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200045</th>\n",
       "      <td>200046</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21440434.0</td>\n",
       "      <td>21441017.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200046</th>\n",
       "      <td>200047</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21481413.0</td>\n",
       "      <td>21482088.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200047</th>\n",
       "      <td>200048</td>\n",
       "      <td>chrY</td>\n",
       "      <td>26408715.0</td>\n",
       "      <td>26409392.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200048</th>\n",
       "      <td>200049</td>\n",
       "      <td>chrY</td>\n",
       "      <td>26476761.0</td>\n",
       "      <td>26477438.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200049 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0   chr       start         end\n",
       "0                1  chr1    797406.0    798178.0\n",
       "1                2  chr1    816037.0    818244.0\n",
       "2                3  chr1    819296.0    820336.0\n",
       "3                4  chr1    826435.0    828307.0\n",
       "4                5  chr1    842686.0    843233.0\n",
       "...            ...   ...         ...         ...\n",
       "200044      200045  chrY  21421806.0  21422749.0\n",
       "200045      200046  chrY  21440434.0  21441017.0\n",
       "200046      200047  chrY  21481413.0  21482088.0\n",
       "200047      200048  chrY  26408715.0  26409392.0\n",
       "200048      200049  chrY  26476761.0  26477438.0\n",
       "\n",
       "[200049 rows x 4 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres = merged_cres.drop(columns=['Unnamed: 0'])\n",
    "merged_cres['chr'] = merged_cres['chr'].str.replace('chr', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres['start'] = merged_cres['start'].astype(int)\n",
    "merged_cres['end'] = merged_cres['end'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cres.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merged CRE TFs table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parse through cre transcription factor file to get 6 columns\n",
    "- tf\n",
    "- merged cre (chr, start, end)\n",
    "- condition\n",
    "- cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres_tfs = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Gene_Mapping/refs/TFBSlog2FCsum_mergedCRE_deAD_sigTF_top0.05dis_neat.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres_tfs.columns = [\n",
    "    merged_cres_tfs.columns[0],  # Keep the first column's name\n",
    "    merged_cres_tfs.columns[1],  # Keep the second column's name\n",
    "    *[re.sub(r'_(.*?)_', '-', col) for col in merged_cres_tfs.columns[2:]]  # Apply the regex to the rest\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>AhrArnt-H1-IFN</th>\n",
       "      <th>Arntl-H1-IFN</th>\n",
       "      <th>BHLHE40-H1-IFN</th>\n",
       "      <th>BHLHE41-H1-IFN</th>\n",
       "      <th>E2F8-H1-IFN</th>\n",
       "      <th>EGR1-H1-IFN</th>\n",
       "      <th>EGR3-H1-IFN</th>\n",
       "      <th>...</th>\n",
       "      <th>ZNF35-iPSC-coculture</th>\n",
       "      <th>ZNF454-iPSC-coculture</th>\n",
       "      <th>ZNF460-iPSC-coculture</th>\n",
       "      <th>ZNF530-iPSC-coculture</th>\n",
       "      <th>ZNF558-iPSC-coculture</th>\n",
       "      <th>ZNF610-iPSC-coculture</th>\n",
       "      <th>ZNF740-iPSC-coculture</th>\n",
       "      <th>ZNF8-iPSC-coculture</th>\n",
       "      <th>ZNF93-iPSC-coculture</th>\n",
       "      <th>Zfp961-iPSC-coculture</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>797406</td>\n",
       "      <td>798178</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>819296</td>\n",
       "      <td>820336</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>842686</td>\n",
       "      <td>843233</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 2106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr   start     end  AhrArnt-H1-IFN  Arntl-H1-IFN  BHLHE40-H1-IFN  \\\n",
       "0  chr1  797406  798178               0             0               0   \n",
       "1  chr1  816037  818244               0             0               0   \n",
       "2  chr1  819296  820336               0             0               0   \n",
       "3  chr1  826435  828307               1             0               0   \n",
       "4  chr1  842686  843233               0             0               0   \n",
       "\n",
       "   BHLHE41-H1-IFN  E2F8-H1-IFN  EGR1-H1-IFN  EGR3-H1-IFN  ...  \\\n",
       "0               0            0            0            0  ...   \n",
       "1               0            0            0            0  ...   \n",
       "2               0            0            0            0  ...   \n",
       "3               0            0            1            1  ...   \n",
       "4               0            0            0            0  ...   \n",
       "\n",
       "   ZNF35-iPSC-coculture  ZNF454-iPSC-coculture  ZNF460-iPSC-coculture  \\\n",
       "0                     0                      0                      0   \n",
       "1                     0                      0                      0   \n",
       "2                     0                      0                      0   \n",
       "3                     0                      0                      0   \n",
       "4                     0                      0                      0   \n",
       "\n",
       "   ZNF530-iPSC-coculture  ZNF558-iPSC-coculture  ZNF610-iPSC-coculture  \\\n",
       "0                      0                      0                      0   \n",
       "1                      0                      0                      0   \n",
       "2                      0                      1                      0   \n",
       "3                      0                      0                      0   \n",
       "4                      0                      0                      0   \n",
       "\n",
       "   ZNF740-iPSC-coculture  ZNF8-iPSC-coculture  ZNF93-iPSC-coculture  \\\n",
       "0                      0                    0                     0   \n",
       "1                      0                    0                     0   \n",
       "2                      0                    0                     0   \n",
       "3                      0                    0                     0   \n",
       "4                      0                    0                     0   \n",
       "\n",
       "   Zfp961-iPSC-coculture  \n",
       "0                      0  \n",
       "1                      0  \n",
       "2                      0  \n",
       "3                      0  \n",
       "4                      0  \n",
       "\n",
       "[5 rows x 2106 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cres_tfs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_merged_cres_faster(df):\n",
    "    # Extract columns representing chromosome, start, and end\n",
    "    chr_col = df['chr'].values\n",
    "    start_col = df['start'].values\n",
    "    end_col = df['end'].values\n",
    "    \n",
    "    # Get all column names except 'chr', 'start', 'end'\n",
    "    feature_cols = [col for col in df.columns if col not in ['chr', 'start', 'end']]\n",
    "    \n",
    "    # Prepare lists to collect data\n",
    "    keys = []\n",
    "    values = []\n",
    "    \n",
    "    # Process each feature column\n",
    "    for col_name in feature_cols:\n",
    "        # Find indices where value is 1\n",
    "        indices = np.where(df[col_name].values == 1)[0]\n",
    "        \n",
    "        if len(indices) > 0:\n",
    "            names = col_name.split('-')\n",
    "            \n",
    "            # Prepare the cell and condition values based on column name pattern\n",
    "            if len(names) == 2:\n",
    "                condition = names[1]\n",
    "                if condition in ['SORL1A528T', 'SORL1KO', 'TREM2R47H', 'TREM2KO', 'CD33', 'INPP5D']:\n",
    "                    cell = 'ESC'\n",
    "                elif condition in ['APOE4vs2', 'APOE4vs3', 'APOEKOvs2', 'APOEKOvs3', 'HIVactivated', 'HIVlatent']:\n",
    "                    cell = 'iPSC'\n",
    "                else:\n",
    "                    cell = 'Unknown'\n",
    "                tf, cell_type, cond = names[0], cell, condition\n",
    "            elif len(names) == 3:\n",
    "                cell = names[1]\n",
    "                condition = names[2]\n",
    "                if condition in ['ACTvsLAT', 'CRISPR','ACSL1']:\n",
    "                    cell = 'iPSC'\n",
    "                    condition = f'{names[1]}-{names[2]}'\n",
    "                tf, cell_type, cond = names[0], cell, condition\n",
    "            elif len(names) == 4:\n",
    "                tf, cell_type, cond = f'{names[0]}-{names[1]}', names[2], names[3]\n",
    "            else:\n",
    "                continue  # Skip if pattern doesn't match\n",
    "            \n",
    "            # For each index where value is 1, create a record\n",
    "            for idx in indices:\n",
    "                keys.append(f'{idx}_{col_name}')\n",
    "                values.append([chr_col[idx], start_col[idx], end_col[idx], tf, cell_type, cond])\n",
    "    \n",
    "    # Create dataframe from collected data\n",
    "    result_df = pd.DataFrame(values, index=keys, \n",
    "                            columns=[0, 1, 2, 3, 4, 5])\n",
    "    \n",
    "    return result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cres_tfs_trial = merged_cres_tfs.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1_Bach1Mafk-WTC11-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>Bach1Mafk</td>\n",
       "      <td>WTC11</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_Erg-WTC11-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>Erg</td>\n",
       "      <td>WTC11</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_GABPA-WTC11-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>GABPA</td>\n",
       "      <td>WTC11</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_IKZF2-WTC11-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>IKZF2</td>\n",
       "      <td>WTC11</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_IRF2-WTC11-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>IRF2</td>\n",
       "      <td>WTC11</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_Nrf1-iPSC-coculture</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>Nrf1</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_SP3-iPSC-coculture</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>SP3</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_SPIB-iPSC-coculture</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>SPIB</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_SPIC-iPSC-coculture</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>SPIC</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_ZNF354A-iPSC-coculture</th>\n",
       "      <td>chr1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "      <td>ZNF354A</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             0       1       2          3      4          5\n",
       "1_Bach1Mafk-WTC11-IFN     chr1  816037  818244  Bach1Mafk  WTC11        IFN\n",
       "1_Erg-WTC11-IFN           chr1  816037  818244        Erg  WTC11        IFN\n",
       "1_GABPA-WTC11-IFN         chr1  816037  818244      GABPA  WTC11        IFN\n",
       "1_IKZF2-WTC11-IFN         chr1  816037  818244      IKZF2  WTC11        IFN\n",
       "1_IRF2-WTC11-IFN          chr1  816037  818244       IRF2  WTC11        IFN\n",
       "...                        ...     ...     ...        ...    ...        ...\n",
       "1_Nrf1-iPSC-coculture     chr1  816037  818244       Nrf1   iPSC  coculture\n",
       "1_SP3-iPSC-coculture      chr1  816037  818244        SP3   iPSC  coculture\n",
       "1_SPIB-iPSC-coculture     chr1  816037  818244       SPIB   iPSC  coculture\n",
       "1_SPIC-iPSC-coculture     chr1  816037  818244       SPIC   iPSC  coculture\n",
       "1_ZNF354A-iPSC-coculture  chr1  816037  818244    ZNF354A   iPSC  coculture\n",
       "\n",
       "[243 rows x 6 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parse_merged_cres_faster(merged_cres_tfs_trial)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedcre_tf = parse_merged_cres_faster(merged_cres_tfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedcre_tf.columns = ['chr', 'start', 'end', 'transcription_factor', 'cell_type', 'condition']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3_AhrArnt-H1-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>H1</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94_AhrArnt-H1-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1397971</td>\n",
       "      <td>1400475</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>H1</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124_AhrArnt-H1-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1614037</td>\n",
       "      <td>1616691</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>H1</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142_AhrArnt-H1-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1745525</td>\n",
       "      <td>1747386</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>H1</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218_AhrArnt-H1-IFN</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2226872</td>\n",
       "      <td>2228282</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>H1</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199972_Zfp961-iPSC-coculture</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13751202</td>\n",
       "      <td>13752042</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199988_Zfp961-iPSC-coculture</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14523314</td>\n",
       "      <td>14525373</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199992_Zfp961-iPSC-coculture</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14840305</td>\n",
       "      <td>14840848</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200025_Zfp961-iPSC-coculture</th>\n",
       "      <td>chrY</td>\n",
       "      <td>19075637</td>\n",
       "      <td>19076488</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200030_Zfp961-iPSC-coculture</th>\n",
       "      <td>chrY</td>\n",
       "      <td>19566739</td>\n",
       "      <td>19568170</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18459792 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               chr     start       end transcription_factor  \\\n",
       "3_AhrArnt-H1-IFN              chr1    826435    828307              AhrArnt   \n",
       "94_AhrArnt-H1-IFN             chr1   1397971   1400475              AhrArnt   \n",
       "124_AhrArnt-H1-IFN            chr1   1614037   1616691              AhrArnt   \n",
       "142_AhrArnt-H1-IFN            chr1   1745525   1747386              AhrArnt   \n",
       "218_AhrArnt-H1-IFN            chr1   2226872   2228282              AhrArnt   \n",
       "...                            ...       ...       ...                  ...   \n",
       "199972_Zfp961-iPSC-coculture  chrY  13751202  13752042               Zfp961   \n",
       "199988_Zfp961-iPSC-coculture  chrY  14523314  14525373               Zfp961   \n",
       "199992_Zfp961-iPSC-coculture  chrY  14840305  14840848               Zfp961   \n",
       "200025_Zfp961-iPSC-coculture  chrY  19075637  19076488               Zfp961   \n",
       "200030_Zfp961-iPSC-coculture  chrY  19566739  19568170               Zfp961   \n",
       "\n",
       "                             cell_type  condition  \n",
       "3_AhrArnt-H1-IFN                    H1        IFN  \n",
       "94_AhrArnt-H1-IFN                   H1        IFN  \n",
       "124_AhrArnt-H1-IFN                  H1        IFN  \n",
       "142_AhrArnt-H1-IFN                  H1        IFN  \n",
       "218_AhrArnt-H1-IFN                  H1        IFN  \n",
       "...                                ...        ...  \n",
       "199972_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199988_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199992_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200025_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200030_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "\n",
       "[18459792 rows x 6 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mergedcre_tf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedcre_tf['cell_type'] = mergedcre_tf['cell_type'].replace('H1', 'ESC')\n",
    "mergedcre_tf['cell_type'] = mergedcre_tf['cell_type'].replace('WTC11', 'iPSC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergedcre_tf['chr'] = mergedcre_tf['chr'].str.replace('chr', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1397971</td>\n",
       "      <td>1400475</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1614037</td>\n",
       "      <td>1616691</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1745525</td>\n",
       "      <td>1747386</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>2226872</td>\n",
       "      <td>2228282</td>\n",
       "      <td>AhrArnt</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199972_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>13751202</td>\n",
       "      <td>13752042</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199988_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>14523314</td>\n",
       "      <td>14525373</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199992_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840305</td>\n",
       "      <td>14840848</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200025_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>19075637</td>\n",
       "      <td>19076488</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200030_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>19566739</td>\n",
       "      <td>19568170</td>\n",
       "      <td>Zfp961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18459792 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             chr     start       end transcription_factor  \\\n",
       "3_AhrArnt-H1-IFN               1    826435    828307              AhrArnt   \n",
       "94_AhrArnt-H1-IFN              1   1397971   1400475              AhrArnt   \n",
       "124_AhrArnt-H1-IFN             1   1614037   1616691              AhrArnt   \n",
       "142_AhrArnt-H1-IFN             1   1745525   1747386              AhrArnt   \n",
       "218_AhrArnt-H1-IFN             1   2226872   2228282              AhrArnt   \n",
       "...                           ..       ...       ...                  ...   \n",
       "199972_Zfp961-iPSC-coculture   Y  13751202  13752042               Zfp961   \n",
       "199988_Zfp961-iPSC-coculture   Y  14523314  14525373               Zfp961   \n",
       "199992_Zfp961-iPSC-coculture   Y  14840305  14840848               Zfp961   \n",
       "200025_Zfp961-iPSC-coculture   Y  19075637  19076488               Zfp961   \n",
       "200030_Zfp961-iPSC-coculture   Y  19566739  19568170               Zfp961   \n",
       "\n",
       "                             cell_type  condition  \n",
       "3_AhrArnt-H1-IFN                   ESC        IFN  \n",
       "94_AhrArnt-H1-IFN                  ESC        IFN  \n",
       "124_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "142_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "218_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "...                                ...        ...  \n",
       "199972_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199988_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199992_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200025_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200030_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "\n",
       "[18459792 rows x 6 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mergedcre_tf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter wanted conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['IFN', 'SORL1A528T', 'SORL1KO', 'TREM2R47H', 'TREM2KO', 'CD33',\n",
       "       'INPP5D', 'APOE3-ACSL1', 'APOE4vs2', 'APOE4vs3', 'APOEKOvs2',\n",
       "       'APOEKOvs3', 'CLU-CRISPR', 'HIVactivated', 'HIV-ACTvsLAT',\n",
       "       'HIVlatent', 'xenot8w', 'xenot12d', 'xenot7d', 'coculture'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mergedcre_tf['condition'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       condition chr     start       end transcription_factor cell_type\n",
      "0    APOE3-ACSL1   1    816037    818244               Bcl11B      iPSC\n",
      "1       APOE4vs2   1    826435    828307              AhrArnt      iPSC\n",
      "2       APOE4vs3   1   2289848   2291601                 Alx1      iPSC\n",
      "3      APOEKOvs2   1    826435    828307              AhrArnt      iPSC\n",
      "4      APOEKOvs3   1    826435    828307              AhrArnt      iPSC\n",
      "5           CD33   1   1157014   1158958               Arid3a       ESC\n",
      "6     CLU-CRISPR   1  17128174  17129001                 Alx1      iPSC\n",
      "7   HIV-ACTvsLAT   1    869480    870292                ASCL1      iPSC\n",
      "8   HIVactivated   1    937942    938810                 Atf3      iPSC\n",
      "9      HIVlatent   1    869480    870292                ASCL1      iPSC\n",
      "10           IFN   1    826435    828307              AhrArnt       ESC\n",
      "11        INPP5D   1   8702524   8704123               Arid3b       ESC\n",
      "12    SORL1A528T   1    826435    828307                ASCL1       ESC\n",
      "13       SORL1KO   1   1230873   1232674                ARNT2       ESC\n",
      "14       TREM2KO   1   1230873   1232674                ARNT2       ESC\n",
      "15     TREM2R47H   1   7478521   7479594               Arid3b       ESC\n",
      "16     coculture   1    826435    828307              AhrArnt      iPSC\n",
      "17      xenot12d   1    816037    818244               Bcl11B      iPSC\n",
      "18       xenot7d   1   6212269   6213348                ARGFX      iPSC\n",
      "19       xenot8w   1    920808    921599                ARNT2      iPSC\n"
     ]
    }
   ],
   "source": [
    "first_rows = mergedcre_tf.groupby('condition', as_index=False).first()\n",
    "print(first_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = mergedcre_tf[mergedcre_tf['condition'].isin(['IFN', 'TREM2R47H', 'TREM2KO', 'xenot7d', 'coculture'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y_/tfywlws16k71l2k62r0j683m0000gn/T/ipykernel_7367/3909949217.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_df['transcription_factor'] = merged_df['transcription_factor'].str.upper()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1397971</td>\n",
       "      <td>1400475</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1614037</td>\n",
       "      <td>1616691</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>1745525</td>\n",
       "      <td>1747386</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218_AhrArnt-H1-IFN</th>\n",
       "      <td>1</td>\n",
       "      <td>2226872</td>\n",
       "      <td>2228282</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199972_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>13751202</td>\n",
       "      <td>13752042</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199988_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>14523314</td>\n",
       "      <td>14525373</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199992_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840305</td>\n",
       "      <td>14840848</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200025_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>19075637</td>\n",
       "      <td>19076488</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200030_Zfp961-iPSC-coculture</th>\n",
       "      <td>Y</td>\n",
       "      <td>19566739</td>\n",
       "      <td>19568170</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5686434 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             chr     start       end transcription_factor  \\\n",
       "3_AhrArnt-H1-IFN               1    826435    828307              AHRARNT   \n",
       "94_AhrArnt-H1-IFN              1   1397971   1400475              AHRARNT   \n",
       "124_AhrArnt-H1-IFN             1   1614037   1616691              AHRARNT   \n",
       "142_AhrArnt-H1-IFN             1   1745525   1747386              AHRARNT   \n",
       "218_AhrArnt-H1-IFN             1   2226872   2228282              AHRARNT   \n",
       "...                           ..       ...       ...                  ...   \n",
       "199972_Zfp961-iPSC-coculture   Y  13751202  13752042               ZFP961   \n",
       "199988_Zfp961-iPSC-coculture   Y  14523314  14525373               ZFP961   \n",
       "199992_Zfp961-iPSC-coculture   Y  14840305  14840848               ZFP961   \n",
       "200025_Zfp961-iPSC-coculture   Y  19075637  19076488               ZFP961   \n",
       "200030_Zfp961-iPSC-coculture   Y  19566739  19568170               ZFP961   \n",
       "\n",
       "                             cell_type  condition  \n",
       "3_AhrArnt-H1-IFN                   ESC        IFN  \n",
       "94_AhrArnt-H1-IFN                  ESC        IFN  \n",
       "124_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "142_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "218_AhrArnt-H1-IFN                 ESC        IFN  \n",
       "...                                ...        ...  \n",
       "199972_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199988_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "199992_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200025_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "200030_Zfp961-iPSC-coculture      iPSC  coculture  \n",
       "\n",
       "[5686434 rows x 6 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['transcription_factor'] = merged_df['transcription_factor'].str.upper()\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cre_tf.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transcription Factors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcription_factor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AHRARNT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARNTL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BHLHE40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BHLHE41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>E2F8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>NKX6-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>MIX-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>CDX1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>NFATC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524</th>\n",
       "      <td>NFATC2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>525 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    transcription_factor\n",
       "0                AHRARNT\n",
       "1                  ARNTL\n",
       "2                BHLHE40\n",
       "3                BHLHE41\n",
       "4                   E2F8\n",
       "..                   ...\n",
       "520               NKX6-1\n",
       "521                MIX-A\n",
       "522                 CDX1\n",
       "523               NFATC4\n",
       "524               NFATC2\n",
       "\n",
       "[525 rows x 1 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tfs = pd.DataFrame(mergedcre_tf['transcription_factor'].unique(), columns=['transcription_factor'])\n",
    "tfs['transcription_factor'] = tfs['transcription_factor'].str.upper()\n",
    "tfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "tfs.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/tfs.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SORL1A528T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SORL1KO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TREM2R47H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TREM2KO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CD33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>INPP5D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>APOE3-ACSL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>APOE4vs2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>APOE4vs3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>APOEKOvs2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>APOEKOvs3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CLU-CRISPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HIVactivated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HIV-ACTvsLAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HIVlatent</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>xenot8w</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>xenot12d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>xenot7d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       condition\n",
       "0            IFN\n",
       "1     SORL1A528T\n",
       "2        SORL1KO\n",
       "3      TREM2R47H\n",
       "4        TREM2KO\n",
       "5           CD33\n",
       "6         INPP5D\n",
       "7    APOE3-ACSL1\n",
       "8       APOE4vs2\n",
       "9       APOE4vs3\n",
       "10     APOEKOvs2\n",
       "11     APOEKOvs3\n",
       "12    CLU-CRISPR\n",
       "13  HIVactivated\n",
       "14  HIV-ACTvsLAT\n",
       "15     HIVlatent\n",
       "16       xenot8w\n",
       "17      xenot12d\n",
       "18       xenot7d\n",
       "19     coculture"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conditions = pd.DataFrame(mergedcre_tf['condition'].unique(), columns=['condition'])\n",
    "conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "#conditions['disease'] = 'AD'\n",
    "conditions.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/conditions.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  cell_type\n",
       "0       ESC\n",
       "1      iPSC\n",
       "2         1"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell = pd.DataFrame(mergedcre_tf['cell_type'].unique(), columns=['cell_type'])\n",
    "cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "cell.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/cell_type.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IFIT2</td>\n",
       "      <td>ENSG00000119922</td>\n",
       "      <td>3433</td>\n",
       "      <td>10</td>\n",
       "      <td>89283694.0</td>\n",
       "      <td>89309271.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CXCL10</td>\n",
       "      <td>ENSG00000169245</td>\n",
       "      <td>3627</td>\n",
       "      <td>4</td>\n",
       "      <td>76021118.0</td>\n",
       "      <td>76023497.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RSAD2</td>\n",
       "      <td>ENSG00000134321</td>\n",
       "      <td>91543</td>\n",
       "      <td>2</td>\n",
       "      <td>6865557.0</td>\n",
       "      <td>6898239.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CXCL11</td>\n",
       "      <td>ENSG00000169248</td>\n",
       "      <td>6373</td>\n",
       "      <td>4</td>\n",
       "      <td>76033682.0</td>\n",
       "      <td>76041415.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TNFSF10</td>\n",
       "      <td>ENSG00000121858</td>\n",
       "      <td>8743</td>\n",
       "      <td>3</td>\n",
       "      <td>172505508.0</td>\n",
       "      <td>172523475.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23416</th>\n",
       "      <td>112268317</td>\n",
       "      <td>ENSG00000307470</td>\n",
       "      <td>112268317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23417</th>\n",
       "      <td>107987399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>107987399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23418</th>\n",
       "      <td>105379561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>105379561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23419</th>\n",
       "      <td>112267916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112267916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23420</th>\n",
       "      <td>107987401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>107987401</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23420 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            hgnc  ensembl_gene_id     entrez chromosome  start_position  \\\n",
       "1          IFIT2  ENSG00000119922       3433         10      89283694.0   \n",
       "2         CXCL10  ENSG00000169245       3627          4      76021118.0   \n",
       "3          RSAD2  ENSG00000134321      91543          2       6865557.0   \n",
       "4         CXCL11  ENSG00000169248       6373          4      76033682.0   \n",
       "5        TNFSF10  ENSG00000121858       8743          3     172505508.0   \n",
       "...          ...              ...        ...        ...             ...   \n",
       "23416  112268317  ENSG00000307470  112268317        NaN             NaN   \n",
       "23417  107987399              NaN  107987399        NaN             NaN   \n",
       "23418  105379561              NaN  105379561        NaN             NaN   \n",
       "23419  112267916              NaN  112267916        NaN             NaN   \n",
       "23420  107987401              NaN  107987401        NaN             NaN   \n",
       "\n",
       "       end_position strand  \n",
       "1        89309271.0      +  \n",
       "2        76023497.0      -  \n",
       "3         6898239.0      +  \n",
       "4        76041415.0      -  \n",
       "5       172523475.0      -  \n",
       "...             ...    ...  \n",
       "23416           NaN    NaN  \n",
       "23417           NaN    NaN  \n",
       "23418           NaN    NaN  \n",
       "23419           NaN    NaN  \n",
       "23420           NaN    NaN  \n",
       "\n",
       "[23420 rows x 7 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_inf = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Genes/ifnb_wtc11_genes.csv', index_col=0)\n",
    "genes_inf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>ENSG00000275199</td>\n",
       "      <td>10000</td>\n",
       "      <td>HSCHR1_3_CTG32_1</td>\n",
       "      <td>500341</td>\n",
       "      <td>867542</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>ENSG00000117020</td>\n",
       "      <td>10000</td>\n",
       "      <td>1</td>\n",
       "      <td>243488233</td>\n",
       "      <td>243851079</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SNORA36C</td>\n",
       "      <td>ENSG00000207016</td>\n",
       "      <td>100124535</td>\n",
       "      <td>2</td>\n",
       "      <td>69520043</td>\n",
       "      <td>69520174</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SNORA70B</td>\n",
       "      <td>ENSG00000206937</td>\n",
       "      <td>100124537</td>\n",
       "      <td>2</td>\n",
       "      <td>61417244</td>\n",
       "      <td>61417378</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIR216B</td>\n",
       "      <td>ENSG00000211520</td>\n",
       "      <td>100126319</td>\n",
       "      <td>2</td>\n",
       "      <td>56000714</td>\n",
       "      <td>56000795</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30363</th>\n",
       "      <td>RBX1</td>\n",
       "      <td>ENSG00000100387</td>\n",
       "      <td>9978</td>\n",
       "      <td>22</td>\n",
       "      <td>40951347</td>\n",
       "      <td>40973309</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30364</th>\n",
       "      <td>DOP1B</td>\n",
       "      <td>ENSG00000142197</td>\n",
       "      <td>9980</td>\n",
       "      <td>21</td>\n",
       "      <td>36156782</td>\n",
       "      <td>36294274</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30365</th>\n",
       "      <td>KCNE2</td>\n",
       "      <td>ENSG00000159197</td>\n",
       "      <td>9992</td>\n",
       "      <td>21</td>\n",
       "      <td>34180729</td>\n",
       "      <td>34371381</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30366</th>\n",
       "      <td>DGCR2</td>\n",
       "      <td>ENSG00000070413</td>\n",
       "      <td>9993</td>\n",
       "      <td>22</td>\n",
       "      <td>19036282</td>\n",
       "      <td>19122454</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30367</th>\n",
       "      <td>SCO2</td>\n",
       "      <td>ENSG00000284194</td>\n",
       "      <td>9997</td>\n",
       "      <td>22</td>\n",
       "      <td>50523568</td>\n",
       "      <td>50526461</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30368 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           hgnc  ensembl_gene_id     entrez        chromosome  start_position  \\\n",
       "0          AKT3  ENSG00000275199      10000  HSCHR1_3_CTG32_1          500341   \n",
       "1          AKT3  ENSG00000117020      10000                 1       243488233   \n",
       "2      SNORA36C  ENSG00000207016  100124535                 2        69520043   \n",
       "3      SNORA70B  ENSG00000206937  100124537                 2        61417244   \n",
       "4       MIR216B  ENSG00000211520  100126319                 2        56000714   \n",
       "...         ...              ...        ...               ...             ...   \n",
       "30363      RBX1  ENSG00000100387       9978                22        40951347   \n",
       "30364     DOP1B  ENSG00000142197       9980                21        36156782   \n",
       "30365     KCNE2  ENSG00000159197       9992                21        34180729   \n",
       "30366     DGCR2  ENSG00000070413       9993                22        19036282   \n",
       "30367      SCO2  ENSG00000284194       9997                22        50523568   \n",
       "\n",
       "       end_position strand  \n",
       "0            867542      -  \n",
       "1         243851079      -  \n",
       "2          69520174      -  \n",
       "3          61417378      -  \n",
       "4          56000795      -  \n",
       "...             ...    ...  \n",
       "30363      40973309      +  \n",
       "30364      36294274      +  \n",
       "30365      34371381      +  \n",
       "30366      19122454      -  \n",
       "30367      50526461      -  \n",
       "\n",
       "[30368 rows x 7 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_trem2 = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Genes/trem2_genes.csv')\n",
    "genes_trem2 = genes_trem2[['HGNC Symbol', 'Ensembl ID', 'Entrez ID', 'Chromosome', 'Start', 'End', 'Strand']]\n",
    "genes_trem2.columns = ['hgnc', 'ensembl_gene_id', 'entrez', 'chromosome', 'start_position', 'end_position', 'strand']\n",
    "genes_trem2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADA</td>\n",
       "      <td>ENSG00000196839</td>\n",
       "      <td>100</td>\n",
       "      <td>20</td>\n",
       "      <td>44584896</td>\n",
       "      <td>44652252</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CDH2</td>\n",
       "      <td>ENSG00000170558</td>\n",
       "      <td>1000</td>\n",
       "      <td>18</td>\n",
       "      <td>27932879</td>\n",
       "      <td>28177946</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RNA5-8SN5</td>\n",
       "      <td>ENSG00000274917</td>\n",
       "      <td>100008587</td>\n",
       "      <td>GL000220.1</td>\n",
       "      <td>112025</td>\n",
       "      <td>112177</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRA1</td>\n",
       "      <td>ENSG00000213523</td>\n",
       "      <td>10011</td>\n",
       "      <td>5</td>\n",
       "      <td>140537340</td>\n",
       "      <td>140557677</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RANBP3-DT</td>\n",
       "      <td>ENSG00000266983</td>\n",
       "      <td>100128568</td>\n",
       "      <td>19</td>\n",
       "      <td>5978176</td>\n",
       "      <td>6020363</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1268</th>\n",
       "      <td>KBTBD11</td>\n",
       "      <td>ENSG00000273645</td>\n",
       "      <td>9920</td>\n",
       "      <td>HSCHR8_8_CTG1</td>\n",
       "      <td>157254</td>\n",
       "      <td>190312</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>KBTBD11</td>\n",
       "      <td>ENSG00000176595</td>\n",
       "      <td>9920</td>\n",
       "      <td>8</td>\n",
       "      <td>1973677</td>\n",
       "      <td>2006936</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>PAN2</td>\n",
       "      <td>ENSG00000135473</td>\n",
       "      <td>9924</td>\n",
       "      <td>12</td>\n",
       "      <td>56316223</td>\n",
       "      <td>56334053</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>KIF14</td>\n",
       "      <td>ENSG00000118193</td>\n",
       "      <td>9928</td>\n",
       "      <td>1</td>\n",
       "      <td>200551497</td>\n",
       "      <td>200620751</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1272</th>\n",
       "      <td>HS3ST3A1</td>\n",
       "      <td>ENSG00000153976</td>\n",
       "      <td>9955</td>\n",
       "      <td>17</td>\n",
       "      <td>13494032</td>\n",
       "      <td>13601929</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1273 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           hgnc  ensembl_gene_id     entrez     chromosome  start_position  \\\n",
       "0           ADA  ENSG00000196839        100             20        44584896   \n",
       "1          CDH2  ENSG00000170558       1000             18        27932879   \n",
       "2     RNA5-8SN5  ENSG00000274917  100008587     GL000220.1          112025   \n",
       "3          SRA1  ENSG00000213523      10011              5       140537340   \n",
       "4     RANBP3-DT  ENSG00000266983  100128568             19         5978176   \n",
       "...         ...              ...        ...            ...             ...   \n",
       "1268    KBTBD11  ENSG00000273645       9920  HSCHR8_8_CTG1          157254   \n",
       "1269    KBTBD11  ENSG00000176595       9920              8         1973677   \n",
       "1270       PAN2  ENSG00000135473       9924             12        56316223   \n",
       "1271      KIF14  ENSG00000118193       9928              1       200551497   \n",
       "1272   HS3ST3A1  ENSG00000153976       9955             17        13494032   \n",
       "\n",
       "      end_position strand  \n",
       "0         44652252      -  \n",
       "1         28177946      -  \n",
       "2           112177      +  \n",
       "3        140557677      -  \n",
       "4          6020363      +  \n",
       "...            ...    ...  \n",
       "1268        190312      +  \n",
       "1269       2006936      +  \n",
       "1270      56334053      -  \n",
       "1271     200620751      -  \n",
       "1272      13601929      -  \n",
       "\n",
       "[1273 rows x 7 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_coculture = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Genes/coculture_genes.csv')\n",
    "genes_coculture['strand'] = genes_coculture['strand'].replace({1: '+', -1: '-'})\n",
    "genes_coculture"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>58345178</td>\n",
       "      <td>58353492</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ADA</td>\n",
       "      <td>ENSG00000196839</td>\n",
       "      <td>100</td>\n",
       "      <td>20</td>\n",
       "      <td>44584896</td>\n",
       "      <td>44652252</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>ENSG00000275199</td>\n",
       "      <td>10000</td>\n",
       "      <td>HSCHR1_3_CTG32_1</td>\n",
       "      <td>500341</td>\n",
       "      <td>867542</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>ENSG00000117020</td>\n",
       "      <td>10000</td>\n",
       "      <td>1</td>\n",
       "      <td>243488233</td>\n",
       "      <td>243851079</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RNA5-8SN5</td>\n",
       "      <td>ENSG00000274917</td>\n",
       "      <td>100008587</td>\n",
       "      <td>GL000220.1</td>\n",
       "      <td>112025</td>\n",
       "      <td>112177</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4234</th>\n",
       "      <td>THOC1</td>\n",
       "      <td>ENSG00000079134</td>\n",
       "      <td>9984</td>\n",
       "      <td>18</td>\n",
       "      <td>214520</td>\n",
       "      <td>268050</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4235</th>\n",
       "      <td>HNRNPDL</td>\n",
       "      <td>ENSG00000152795</td>\n",
       "      <td>9987</td>\n",
       "      <td>4</td>\n",
       "      <td>82422565</td>\n",
       "      <td>82430462</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4236</th>\n",
       "      <td>CASP8AP2</td>\n",
       "      <td>ENSG00000288475</td>\n",
       "      <td>9994</td>\n",
       "      <td>HG2121_PATCH</td>\n",
       "      <td>25225</td>\n",
       "      <td>72124</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4237</th>\n",
       "      <td>CASP8AP2</td>\n",
       "      <td>ENSG00000118412</td>\n",
       "      <td>9994</td>\n",
       "      <td>6</td>\n",
       "      <td>89829894</td>\n",
       "      <td>89874436</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4238</th>\n",
       "      <td>SCO2</td>\n",
       "      <td>ENSG00000284194</td>\n",
       "      <td>9997</td>\n",
       "      <td>22</td>\n",
       "      <td>50523568</td>\n",
       "      <td>50526461</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4239 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           hgnc  ensembl_gene_id     entrez        chromosome  start_position  \\\n",
       "0          A1BG  ENSG00000121410          1                19        58345178   \n",
       "1           ADA  ENSG00000196839        100                20        44584896   \n",
       "2          AKT3  ENSG00000275199      10000  HSCHR1_3_CTG32_1          500341   \n",
       "3          AKT3  ENSG00000117020      10000                 1       243488233   \n",
       "4     RNA5-8SN5  ENSG00000274917  100008587        GL000220.1          112025   \n",
       "...         ...              ...        ...               ...             ...   \n",
       "4234      THOC1  ENSG00000079134       9984                18          214520   \n",
       "4235    HNRNPDL  ENSG00000152795       9987                 4        82422565   \n",
       "4236   CASP8AP2  ENSG00000288475       9994      HG2121_PATCH           25225   \n",
       "4237   CASP8AP2  ENSG00000118412       9994                 6        89829894   \n",
       "4238       SCO2  ENSG00000284194       9997                22        50523568   \n",
       "\n",
       "      end_position strand  \n",
       "0         58353492      -  \n",
       "1         44652252      -  \n",
       "2           867542      -  \n",
       "3        243851079      -  \n",
       "4           112177      +  \n",
       "...            ...    ...  \n",
       "4234        268050      -  \n",
       "4235      82430462      -  \n",
       "4236         72124      +  \n",
       "4237      89874436      +  \n",
       "4238      50526461      -  \n",
       "\n",
       "[4239 rows x 7 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_xmg_7days = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Genes/gene_annotations_iMG_vs_xMG7days.csv')\n",
    "genes_xmg_7days['strand'] = genes_xmg_7days['strand'].replace({1: '+', -1: '-'})\n",
    "genes_xmg_7days = genes_xmg_7days[['hgnc', 'ensembl_gene_id', 'entrez', 'chromosome', 'start_position', 'end_position', 'strand']]\n",
    "genes_xmg_7days"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "combine all rows based on entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFIT2</td>\n",
       "      <td>ENSG00000119922</td>\n",
       "      <td>3433</td>\n",
       "      <td>10</td>\n",
       "      <td>89283694.0</td>\n",
       "      <td>89309271.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CXCL10</td>\n",
       "      <td>ENSG00000169245</td>\n",
       "      <td>3627</td>\n",
       "      <td>4</td>\n",
       "      <td>76021118.0</td>\n",
       "      <td>76023497.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RSAD2</td>\n",
       "      <td>ENSG00000134321</td>\n",
       "      <td>91543</td>\n",
       "      <td>2</td>\n",
       "      <td>6865557.0</td>\n",
       "      <td>6898239.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXCL11</td>\n",
       "      <td>ENSG00000169248</td>\n",
       "      <td>6373</td>\n",
       "      <td>4</td>\n",
       "      <td>76033682.0</td>\n",
       "      <td>76041415.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TNFSF10</td>\n",
       "      <td>ENSG00000121858</td>\n",
       "      <td>8743</td>\n",
       "      <td>3</td>\n",
       "      <td>172505508.0</td>\n",
       "      <td>172523475.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53738</th>\n",
       "      <td>PAGE4</td>\n",
       "      <td>ENSG00000101951</td>\n",
       "      <td>9506</td>\n",
       "      <td>X</td>\n",
       "      <td>49793653.0</td>\n",
       "      <td>49834264.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53741</th>\n",
       "      <td>GAL3ST1</td>\n",
       "      <td>ENSG00000128242</td>\n",
       "      <td>9514</td>\n",
       "      <td>22</td>\n",
       "      <td>30554635.0</td>\n",
       "      <td>30574665.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53747</th>\n",
       "      <td>CD40LG</td>\n",
       "      <td>ENSG00000102245</td>\n",
       "      <td>959</td>\n",
       "      <td>X</td>\n",
       "      <td>136648158.0</td>\n",
       "      <td>136660390.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53762</th>\n",
       "      <td>FRMPD4</td>\n",
       "      <td>ENSG00000169933</td>\n",
       "      <td>9758</td>\n",
       "      <td>X</td>\n",
       "      <td>11822423.0</td>\n",
       "      <td>12724523.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53779</th>\n",
       "      <td>MAGEC1</td>\n",
       "      <td>ENSG00000155495</td>\n",
       "      <td>9947</td>\n",
       "      <td>X</td>\n",
       "      <td>141903894.0</td>\n",
       "      <td>141909374.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31020 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          hgnc  ensembl_gene_id  entrez chromosome  start_position  \\\n",
       "0        IFIT2  ENSG00000119922    3433         10      89283694.0   \n",
       "1       CXCL10  ENSG00000169245    3627          4      76021118.0   \n",
       "2        RSAD2  ENSG00000134321   91543          2       6865557.0   \n",
       "3       CXCL11  ENSG00000169248    6373          4      76033682.0   \n",
       "4      TNFSF10  ENSG00000121858    8743          3     172505508.0   \n",
       "...        ...              ...     ...        ...             ...   \n",
       "53738    PAGE4  ENSG00000101951    9506          X      49793653.0   \n",
       "53741  GAL3ST1  ENSG00000128242    9514         22      30554635.0   \n",
       "53747   CD40LG  ENSG00000102245     959          X     136648158.0   \n",
       "53762   FRMPD4  ENSG00000169933    9758          X      11822423.0   \n",
       "53779   MAGEC1  ENSG00000155495    9947          X     141903894.0   \n",
       "\n",
       "       end_position strand  \n",
       "0        89309271.0      +  \n",
       "1        76023497.0      -  \n",
       "2         6898239.0      +  \n",
       "3        76041415.0      -  \n",
       "4       172523475.0      -  \n",
       "...             ...    ...  \n",
       "53738    49834264.0      +  \n",
       "53741    30574665.0      -  \n",
       "53747   136660390.0      +  \n",
       "53762    12724523.0      +  \n",
       "53779   141909374.0      +  \n",
       "\n",
       "[31020 rows x 7 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine all the dfs based on unique entrez\n",
    "combined_genes = pd.concat([genes_inf, genes_trem2, genes_coculture, genes_xmg_7days], ignore_index=True)\n",
    "combined_genes = combined_genes.drop_duplicates(subset=['entrez'])\n",
    "combined_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_genes.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/combined_genes.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential Expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "DE: Combine all differential expression data tables, add in two columns for cell and condition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns in differential_expression_results_iMG_vs_oMG.csv:\n",
      "['ensembl_gene_id', 'geneID', 'gene_symbol', 'gene_name', 'log2FoldChange', 'pvalue', 'padj', 'baseMean']\n",
      "----------------------------------------\n",
      "Columns in TREM2_KO_deseq.csv:\n",
      "['geneID', 'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj', 'gene_symbol', 'description']\n",
      "----------------------------------------\n",
      "Columns in TREM2_R47H_deseq.csv:\n",
      "['geneID', 'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj', 'gene_symbol', 'description']\n",
      "----------------------------------------\n",
      "Columns in WTC11_ifnb_DE.csv:\n",
      "['Unnamed: 0', 'entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
      "----------------------------------------\n",
      "Columns in differential_expression_results_iMG_vs_xMG-7days.csv:\n",
      "['ensembl_gene_id', 'geneID', 'gene_symbol', 'gene_name', 'log2FoldChange', 'pvalue', 'padj', 'baseMean']\n",
      "----------------------------------------\n",
      "Columns in H1_ifnb_DE.csv:\n",
      "['Unnamed: 0', 'entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
      "----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "csv_dir = '/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE'\n",
    "csv_files = glob.glob(os.path.join(csv_dir, '*.csv'))\n",
    "for file in csv_files:\n",
    "    df = pd.read_csv(file)\n",
    "    print(f\"Columns in {os.path.basename(file)}:\")\n",
    "    print(df.columns.tolist())\n",
    "    print('-' * 40)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3433</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>51363.949143</td>\n",
       "      <td>7.076655</td>\n",
       "      <td>2.431514e-161</td>\n",
       "      <td>3.661131e-157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3627</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>31139.923458</td>\n",
       "      <td>7.208627</td>\n",
       "      <td>7.782382e-159</td>\n",
       "      <td>5.858966e-155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>91543</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>17487.285409</td>\n",
       "      <td>7.078944</td>\n",
       "      <td>1.216976e-151</td>\n",
       "      <td>6.108001e-148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6373</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>8241.154933</td>\n",
       "      <td>8.971610</td>\n",
       "      <td>2.625696e-140</td>\n",
       "      <td>9.883776e-137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8743</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5885.554918</td>\n",
       "      <td>8.436834</td>\n",
       "      <td>4.660559e-135</td>\n",
       "      <td>1.403481e-131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23416</th>\n",
       "      <td>112268317</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>8.925314</td>\n",
       "      <td>0.790385</td>\n",
       "      <td>6.401772e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23417</th>\n",
       "      <td>107987399</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>11.602322</td>\n",
       "      <td>0.674284</td>\n",
       "      <td>6.483830e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23418</th>\n",
       "      <td>105379561</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3.020266</td>\n",
       "      <td>-1.957856</td>\n",
       "      <td>5.170214e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23419</th>\n",
       "      <td>112267916</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>9.796965</td>\n",
       "      <td>-2.741803</td>\n",
       "      <td>1.495176e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23420</th>\n",
       "      <td>107987401</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23420 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          entrez condition cell_type      baseMean  log2FoldChange  \\\n",
       "1           3433       IFN      iPSC  51363.949143        7.076655   \n",
       "2           3627       IFN      iPSC  31139.923458        7.208627   \n",
       "3          91543       IFN      iPSC  17487.285409        7.078944   \n",
       "4           6373       IFN      iPSC   8241.154933        8.971610   \n",
       "5           8743       IFN      iPSC   5885.554918        8.436834   \n",
       "...          ...       ...       ...           ...             ...   \n",
       "23416  112268317       IFN      iPSC      8.925314        0.790385   \n",
       "23417  107987399       IFN      iPSC     11.602322        0.674284   \n",
       "23418  105379561       IFN      iPSC      3.020266       -1.957856   \n",
       "23419  112267916       IFN      iPSC      9.796965       -2.741803   \n",
       "23420  107987401       IFN      iPSC      0.000000             NaN   \n",
       "\n",
       "              pvalue           padj  \n",
       "1      2.431514e-161  3.661131e-157  \n",
       "2      7.782382e-159  5.858966e-155  \n",
       "3      1.216976e-151  6.108001e-148  \n",
       "4      2.625696e-140  9.883776e-137  \n",
       "5      4.660559e-135  1.403481e-131  \n",
       "...              ...            ...  \n",
       "23416   6.401772e-01            NaN  \n",
       "23417   6.483830e-01            NaN  \n",
       "23418   5.170214e-01            NaN  \n",
       "23419   1.495176e-01            NaN  \n",
       "23420            NaN            NaN  \n",
       "\n",
       "[23420 rows x 7 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wtc11_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/WTC11_ifnb_DE.csv', index_col=0)\n",
    "wtc11_de.insert(wtc11_de.columns.get_loc('entrez')+1, 'condition', 'IFN')\n",
    "wtc11_de.insert(wtc11_de.columns.get_loc('entrez')+2, 'cell_type', 'iPSC')\n",
    "wtc11_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>91543</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>13475.331282</td>\n",
       "      <td>10.264425</td>\n",
       "      <td>1.447029e-268</td>\n",
       "      <td>2.130895e-264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8638</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>5899.382711</td>\n",
       "      <td>8.398069</td>\n",
       "      <td>1.567318e-220</td>\n",
       "      <td>1.154016e-216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4599</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>16428.403759</td>\n",
       "      <td>6.458007</td>\n",
       "      <td>3.290398e-207</td>\n",
       "      <td>1.615147e-203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3434</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>48908.668496</td>\n",
       "      <td>8.256053</td>\n",
       "      <td>4.550447e-192</td>\n",
       "      <td>1.675247e-188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3437</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>22345.434892</td>\n",
       "      <td>5.937422</td>\n",
       "      <td>1.197989e-182</td>\n",
       "      <td>3.528316e-179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23416</th>\n",
       "      <td>107987401</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>7.769690</td>\n",
       "      <td>-0.399492</td>\n",
       "      <td>7.454384e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23417</th>\n",
       "      <td>4572</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>3.285157</td>\n",
       "      <td>-0.687154</td>\n",
       "      <td>7.282761e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23418</th>\n",
       "      <td>4564</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1.225285</td>\n",
       "      <td>-0.371544</td>\n",
       "      <td>8.955932e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23419</th>\n",
       "      <td>4575</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23420</th>\n",
       "      <td>4568</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>3.832692</td>\n",
       "      <td>-1.874843</td>\n",
       "      <td>2.985244e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23420 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          entrez condition cell_type      baseMean  log2FoldChange  \\\n",
       "1          91543       IFN       ESC  13475.331282       10.264425   \n",
       "2           8638       IFN       ESC   5899.382711        8.398069   \n",
       "3           4599       IFN       ESC  16428.403759        6.458007   \n",
       "4           3434       IFN       ESC  48908.668496        8.256053   \n",
       "5           3437       IFN       ESC  22345.434892        5.937422   \n",
       "...          ...       ...       ...           ...             ...   \n",
       "23416  107987401       IFN       ESC      7.769690       -0.399492   \n",
       "23417       4572       IFN       ESC      3.285157       -0.687154   \n",
       "23418       4564       IFN       ESC      1.225285       -0.371544   \n",
       "23419       4575       IFN       ESC      0.000000             NaN   \n",
       "23420       4568       IFN       ESC      3.832692       -1.874843   \n",
       "\n",
       "              pvalue           padj  \n",
       "1      1.447029e-268  2.130895e-264  \n",
       "2      1.567318e-220  1.154016e-216  \n",
       "3      3.290398e-207  1.615147e-203  \n",
       "4      4.550447e-192  1.675247e-188  \n",
       "5      1.197989e-182  3.528316e-179  \n",
       "...              ...            ...  \n",
       "23416   7.454384e-01            NaN  \n",
       "23417   7.282761e-01            NaN  \n",
       "23418   8.955932e-01            NaN  \n",
       "23419            NaN            NaN  \n",
       "23420   2.985244e-01            NaN  \n",
       "\n",
       "[23420 rows x 7 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h1_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/H1_ifnb_DE.csv', index_col=0)\n",
    "h1_de.insert(h1_de.columns.get_loc('entrez')+1, 'condition', 'IFN')\n",
    "h1_de.insert(h1_de.columns.get_loc('entrez')+2, 'cell_type', 'ESC')\n",
    "h1_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23240</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>390.675732</td>\n",
       "      <td>-0.351666</td>\n",
       "      <td>5.952899e-08</td>\n",
       "      <td>0.000990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>998</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>5228.274617</td>\n",
       "      <td>-0.162124</td>\n",
       "      <td>2.515882e-07</td>\n",
       "      <td>0.002092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2167</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>20.332458</td>\n",
       "      <td>3.446080</td>\n",
       "      <td>4.977782e-07</td>\n",
       "      <td>0.002759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9262</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1589.645549</td>\n",
       "      <td>-0.392097</td>\n",
       "      <td>8.858416e-07</td>\n",
       "      <td>0.003682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8895</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1093.591798</td>\n",
       "      <td>-0.243021</td>\n",
       "      <td>1.999798e-06</td>\n",
       "      <td>0.006651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32815</th>\n",
       "      <td>102723847</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.523504</td>\n",
       "      <td>0.637379</td>\n",
       "      <td>7.002371e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32816</th>\n",
       "      <td>105379417</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.987051</td>\n",
       "      <td>3.401924</td>\n",
       "      <td>5.820441e-02</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32817</th>\n",
       "      <td>105379566</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.195688</td>\n",
       "      <td>0.430377</td>\n",
       "      <td>8.696057e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32818</th>\n",
       "      <td>107987401</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.660971</td>\n",
       "      <td>-0.657456</td>\n",
       "      <td>6.176979e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32819</th>\n",
       "      <td>389834</td>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.996691</td>\n",
       "      <td>0.098813</td>\n",
       "      <td>9.426345e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32820 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          entrez condition cell_type     baseMean  log2FoldChange  \\\n",
       "0          23240   TREM2KO       ESC   390.675732       -0.351666   \n",
       "1            998   TREM2KO       ESC  5228.274617       -0.162124   \n",
       "2           2167   TREM2KO       ESC    20.332458        3.446080   \n",
       "3           9262   TREM2KO       ESC  1589.645549       -0.392097   \n",
       "4           8895   TREM2KO       ESC  1093.591798       -0.243021   \n",
       "...          ...       ...       ...          ...             ...   \n",
       "32815  102723847   TREM2KO       ESC     0.523504        0.637379   \n",
       "32816  105379417   TREM2KO       ESC     0.987051        3.401924   \n",
       "32817  105379566   TREM2KO       ESC     0.195688        0.430377   \n",
       "32818  107987401   TREM2KO       ESC     0.660971       -0.657456   \n",
       "32819     389834   TREM2KO       ESC     0.996691        0.098813   \n",
       "\n",
       "             pvalue      padj  \n",
       "0      5.952899e-08  0.000990  \n",
       "1      2.515882e-07  0.002092  \n",
       "2      4.977782e-07  0.002759  \n",
       "3      8.858416e-07  0.003682  \n",
       "4      1.999798e-06  0.006651  \n",
       "...             ...       ...  \n",
       "32815  7.002371e-01       NaN  \n",
       "32816  5.820441e-02       NaN  \n",
       "32817  8.696057e-01       NaN  \n",
       "32818  6.176979e-01       NaN  \n",
       "32819  9.426345e-01       NaN  \n",
       "\n",
       "[32820 rows x 7 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trem2ko_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/TREM2_KO_deseq.csv')\n",
    "trem2ko_de = trem2ko_de[['geneID', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']]\n",
    "trem2ko_de.columns = ['entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
    "trem2ko_de.insert(trem2ko_de.columns.get_loc('entrez')+1, 'condition', 'TREM2KO')\n",
    "trem2ko_de.insert(trem2ko_de.columns.get_loc('entrez')+2, 'cell_type', 'ESC')\n",
    "trem2ko_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29948</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>18.076160</td>\n",
       "      <td>1.361652</td>\n",
       "      <td>9.298600e-08</td>\n",
       "      <td>0.001378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>64786</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>597.139516</td>\n",
       "      <td>-0.362428</td>\n",
       "      <td>1.413515e-06</td>\n",
       "      <td>0.010471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>84002</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2668.769621</td>\n",
       "      <td>-0.362971</td>\n",
       "      <td>2.633130e-06</td>\n",
       "      <td>0.013003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6461</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>175.548315</td>\n",
       "      <td>0.510790</td>\n",
       "      <td>5.694400e-06</td>\n",
       "      <td>0.016873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>284129</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>217.989695</td>\n",
       "      <td>0.588959</td>\n",
       "      <td>5.190929e-06</td>\n",
       "      <td>0.016873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32815</th>\n",
       "      <td>102723847</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.132067</td>\n",
       "      <td>-0.702108</td>\n",
       "      <td>8.199424e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32816</th>\n",
       "      <td>105379417</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.565925</td>\n",
       "      <td>2.677743</td>\n",
       "      <td>3.765988e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32817</th>\n",
       "      <td>105379566</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.071876</td>\n",
       "      <td>0.534472</td>\n",
       "      <td>8.624372e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32818</th>\n",
       "      <td>107987401</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.507951</td>\n",
       "      <td>1.262275</td>\n",
       "      <td>3.873289e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32819</th>\n",
       "      <td>389834</td>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>0.294062</td>\n",
       "      <td>-0.702106</td>\n",
       "      <td>7.824953e-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32820 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          entrez  condition cell_type     baseMean  log2FoldChange  \\\n",
       "0          29948  TREM2R47H       ESC    18.076160        1.361652   \n",
       "1          64786  TREM2R47H       ESC   597.139516       -0.362428   \n",
       "2          84002  TREM2R47H       ESC  2668.769621       -0.362971   \n",
       "3           6461  TREM2R47H       ESC   175.548315        0.510790   \n",
       "4         284129  TREM2R47H       ESC   217.989695        0.588959   \n",
       "...          ...        ...       ...          ...             ...   \n",
       "32815  102723847  TREM2R47H       ESC     0.132067       -0.702108   \n",
       "32816  105379417  TREM2R47H       ESC     0.565925        2.677743   \n",
       "32817  105379566  TREM2R47H       ESC     0.071876        0.534472   \n",
       "32818  107987401  TREM2R47H       ESC     0.507951        1.262275   \n",
       "32819     389834  TREM2R47H       ESC     0.294062       -0.702106   \n",
       "\n",
       "             pvalue      padj  \n",
       "0      9.298600e-08  0.001378  \n",
       "1      1.413515e-06  0.010471  \n",
       "2      2.633130e-06  0.013003  \n",
       "3      5.694400e-06  0.016873  \n",
       "4      5.190929e-06  0.016873  \n",
       "...             ...       ...  \n",
       "32815  8.199424e-01       NaN  \n",
       "32816  3.765988e-01       NaN  \n",
       "32817  8.624372e-01       NaN  \n",
       "32818  3.873289e-01       NaN  \n",
       "32819  7.824953e-01       NaN  \n",
       "\n",
       "[32820 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trem2r47h_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/TREM2_R47H_deseq.csv')\n",
    "trem2r47h_de = trem2r47h_de[['geneID', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']]\n",
    "trem2r47h_de.columns = ['entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
    "trem2r47h_de.insert(trem2r47h_de.columns.get_loc('entrez')+1, 'condition', 'TREM2R47H')\n",
    "trem2r47h_de.insert(trem2r47h_de.columns.get_loc('entrez')+2, 'cell_type', 'ESC')\n",
    "trem2r47h_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>729737</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>572.381215</td>\n",
       "      <td>-2.117348</td>\n",
       "      <td>5.900259e-04</td>\n",
       "      <td>2.629321e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100132287</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>204.892806</td>\n",
       "      <td>-1.797802</td>\n",
       "      <td>6.878271e-07</td>\n",
       "      <td>6.552458e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100133331</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>181.949972</td>\n",
       "      <td>-1.690312</td>\n",
       "      <td>8.825274e-06</td>\n",
       "      <td>6.415882e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>79854</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>11.200773</td>\n",
       "      <td>-2.080607</td>\n",
       "      <td>6.066192e-03</td>\n",
       "      <td>1.966917e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>643837</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>119.555942</td>\n",
       "      <td>-1.054365</td>\n",
       "      <td>1.436130e-03</td>\n",
       "      <td>5.658187e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4203</th>\n",
       "      <td>4541</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>9611.938348</td>\n",
       "      <td>-1.625844</td>\n",
       "      <td>1.547930e-08</td>\n",
       "      <td>2.126153e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4204</th>\n",
       "      <td>4556</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>465.777487</td>\n",
       "      <td>-1.544319</td>\n",
       "      <td>1.061511e-06</td>\n",
       "      <td>9.720227e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4205</th>\n",
       "      <td>4519</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>28681.291675</td>\n",
       "      <td>-1.450410</td>\n",
       "      <td>4.936545e-08</td>\n",
       "      <td>6.156218e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4206</th>\n",
       "      <td>4576</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>84.263876</td>\n",
       "      <td>1.042201</td>\n",
       "      <td>2.857335e-04</td>\n",
       "      <td>1.388658e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4207</th>\n",
       "      <td>4571</td>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>955.389582</td>\n",
       "      <td>-2.225917</td>\n",
       "      <td>6.247015e-31</td>\n",
       "      <td>1.089963e-28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4208 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         entrez condition cell_type      baseMean  log2FoldChange  \\\n",
       "0        729737   xenot7d      iPSC    572.381215       -2.117348   \n",
       "1     100132287   xenot7d      iPSC    204.892806       -1.797802   \n",
       "2     100133331   xenot7d      iPSC    181.949972       -1.690312   \n",
       "3         79854   xenot7d      iPSC     11.200773       -2.080607   \n",
       "4        643837   xenot7d      iPSC    119.555942       -1.054365   \n",
       "...         ...       ...       ...           ...             ...   \n",
       "4203       4541   xenot7d      iPSC   9611.938348       -1.625844   \n",
       "4204       4556   xenot7d      iPSC    465.777487       -1.544319   \n",
       "4205       4519   xenot7d      iPSC  28681.291675       -1.450410   \n",
       "4206       4576   xenot7d      iPSC     84.263876        1.042201   \n",
       "4207       4571   xenot7d      iPSC    955.389582       -2.225917   \n",
       "\n",
       "            pvalue          padj  \n",
       "0     5.900259e-04  2.629321e-03  \n",
       "1     6.878271e-07  6.552458e-06  \n",
       "2     8.825274e-06  6.415882e-05  \n",
       "3     6.066192e-03  1.966917e-02  \n",
       "4     1.436130e-03  5.658187e-03  \n",
       "...            ...           ...  \n",
       "4203  1.547930e-08  2.126153e-07  \n",
       "4204  1.061511e-06  9.720227e-06  \n",
       "4205  4.936545e-08  6.156218e-07  \n",
       "4206  2.857335e-04  1.388658e-03  \n",
       "4207  6.247015e-31  1.089963e-28  \n",
       "\n",
       "[4208 rows x 7 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xmg_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/differential_expression_results_iMG_vs_xMG-7days.csv')\n",
    "xmg_de = xmg_de[['geneID', 'baseMean','log2FoldChange', 'pvalue', 'padj']] \n",
    "xmg_de.columns = ['entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
    "xmg_de.insert(xmg_de.columns.get_loc('entrez')+1, 'condition', 'xenot7d')\n",
    "xmg_de.insert(xmg_de.columns.get_loc('entrez')+2, 'cell_type', 'iPSC')\n",
    "xmg_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>57801</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3.226657</td>\n",
       "      <td>3.876730</td>\n",
       "      <td>9.604124e-04</td>\n",
       "      <td>1.184946e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>375790</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>64.409736</td>\n",
       "      <td>1.640842</td>\n",
       "      <td>2.947100e-04</td>\n",
       "      <td>4.969108e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>112267874</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3.098629</td>\n",
       "      <td>3.224100</td>\n",
       "      <td>5.706338e-03</td>\n",
       "      <td>4.244529e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>388591</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>70.581749</td>\n",
       "      <td>1.695869</td>\n",
       "      <td>4.529711e-04</td>\n",
       "      <td>6.805862e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>387509</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>13.714961</td>\n",
       "      <td>1.985986</td>\n",
       "      <td>2.471307e-03</td>\n",
       "      <td>2.388710e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>4570</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3017.339959</td>\n",
       "      <td>-1.042412</td>\n",
       "      <td>3.818550e-06</td>\n",
       "      <td>1.901385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>4555</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>74.789086</td>\n",
       "      <td>1.586012</td>\n",
       "      <td>4.269662e-11</td>\n",
       "      <td>1.484118e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>4509</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5001.407250</td>\n",
       "      <td>1.512440</td>\n",
       "      <td>6.821825e-14</td>\n",
       "      <td>4.403732e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1285</th>\n",
       "      <td>4508</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>35394.848958</td>\n",
       "      <td>1.197526</td>\n",
       "      <td>2.689226e-12</td>\n",
       "      <td>1.429640e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1286</th>\n",
       "      <td>4576</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>118.630934</td>\n",
       "      <td>1.638993</td>\n",
       "      <td>8.873869e-11</td>\n",
       "      <td>2.916276e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1287 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         entrez  condition cell_type      baseMean  log2FoldChange  \\\n",
       "0         57801  coculture      iPSC      3.226657        3.876730   \n",
       "1        375790  coculture      iPSC     64.409736        1.640842   \n",
       "2     112267874  coculture      iPSC      3.098629        3.224100   \n",
       "3        388591  coculture      iPSC     70.581749        1.695869   \n",
       "4        387509  coculture      iPSC     13.714961        1.985986   \n",
       "...         ...        ...       ...           ...             ...   \n",
       "1282       4570  coculture      iPSC   3017.339959       -1.042412   \n",
       "1283       4555  coculture      iPSC     74.789086        1.586012   \n",
       "1284       4509  coculture      iPSC   5001.407250        1.512440   \n",
       "1285       4508  coculture      iPSC  35394.848958        1.197526   \n",
       "1286       4576  coculture      iPSC    118.630934        1.638993   \n",
       "\n",
       "            pvalue          padj  \n",
       "0     9.604124e-04  1.184946e-02  \n",
       "1     2.947100e-04  4.969108e-03  \n",
       "2     5.706338e-03  4.244529e-02  \n",
       "3     4.529711e-04  6.805862e-03  \n",
       "4     2.471307e-03  2.388710e-02  \n",
       "...            ...           ...  \n",
       "1282  3.818550e-06  1.901385e-04  \n",
       "1283  4.269662e-11  1.484118e-08  \n",
       "1284  6.821825e-14  4.403732e-11  \n",
       "1285  2.689226e-12  1.429640e-09  \n",
       "1286  8.873869e-11  2.916276e-08  \n",
       "\n",
       "[1287 rows x 7 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omg_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/DE/differential_expression_results.csv')\n",
    "omg_de = omg_de[['geneID', 'baseMean','log2FoldChange', 'pvalue', 'padj']]\n",
    "omg_de.columns = ['entrez', 'baseMean', 'log2FoldChange', 'pvalue', 'padj']\n",
    "omg_de.insert(omg_de.columns.get_loc('entrez')+1, 'condition', 'coculture')\n",
    "omg_de.insert(omg_de.columns.get_loc('entrez')+2, 'cell_type', 'iPSC')\n",
    "omg_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3433</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>51363.949143</td>\n",
       "      <td>7.076655</td>\n",
       "      <td>2.431514e-161</td>\n",
       "      <td>3.661131e-157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3627</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>31139.923458</td>\n",
       "      <td>7.208627</td>\n",
       "      <td>7.782382e-159</td>\n",
       "      <td>5.858966e-155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>91543</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>17487.285409</td>\n",
       "      <td>7.078944</td>\n",
       "      <td>1.216976e-151</td>\n",
       "      <td>6.108001e-148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6373</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>8241.154933</td>\n",
       "      <td>8.971610</td>\n",
       "      <td>2.625696e-140</td>\n",
       "      <td>9.883776e-137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8743</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5885.554918</td>\n",
       "      <td>8.436834</td>\n",
       "      <td>4.660559e-135</td>\n",
       "      <td>1.403481e-131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117970</th>\n",
       "      <td>4570</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3017.339959</td>\n",
       "      <td>-1.042412</td>\n",
       "      <td>3.818550e-06</td>\n",
       "      <td>1.901385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117971</th>\n",
       "      <td>4555</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>74.789086</td>\n",
       "      <td>1.586012</td>\n",
       "      <td>4.269662e-11</td>\n",
       "      <td>1.484118e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117972</th>\n",
       "      <td>4509</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5001.407250</td>\n",
       "      <td>1.512440</td>\n",
       "      <td>6.821825e-14</td>\n",
       "      <td>4.403732e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117973</th>\n",
       "      <td>4508</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>35394.848958</td>\n",
       "      <td>1.197526</td>\n",
       "      <td>2.689226e-12</td>\n",
       "      <td>1.429640e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117974</th>\n",
       "      <td>4576</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>118.630934</td>\n",
       "      <td>1.638993</td>\n",
       "      <td>8.873869e-11</td>\n",
       "      <td>2.916276e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117975 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        entrez  condition cell_type      baseMean  log2FoldChange  \\\n",
       "0         3433        IFN      iPSC  51363.949143        7.076655   \n",
       "1         3627        IFN      iPSC  31139.923458        7.208627   \n",
       "2        91543        IFN      iPSC  17487.285409        7.078944   \n",
       "3         6373        IFN      iPSC   8241.154933        8.971610   \n",
       "4         8743        IFN      iPSC   5885.554918        8.436834   \n",
       "...        ...        ...       ...           ...             ...   \n",
       "117970    4570  coculture      iPSC   3017.339959       -1.042412   \n",
       "117971    4555  coculture      iPSC     74.789086        1.586012   \n",
       "117972    4509  coculture      iPSC   5001.407250        1.512440   \n",
       "117973    4508  coculture      iPSC  35394.848958        1.197526   \n",
       "117974    4576  coculture      iPSC    118.630934        1.638993   \n",
       "\n",
       "               pvalue           padj  \n",
       "0       2.431514e-161  3.661131e-157  \n",
       "1       7.782382e-159  5.858966e-155  \n",
       "2       1.216976e-151  6.108001e-148  \n",
       "3       2.625696e-140  9.883776e-137  \n",
       "4       4.660559e-135  1.403481e-131  \n",
       "...               ...            ...  \n",
       "117970   3.818550e-06   1.901385e-04  \n",
       "117971   4.269662e-11   1.484118e-08  \n",
       "117972   6.821825e-14   4.403732e-11  \n",
       "117973   2.689226e-12   1.429640e-09  \n",
       "117974   8.873869e-11   2.916276e-08  \n",
       "\n",
       "[117975 rows x 7 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combiend_de = pd.concat([wtc11_de, h1_de, trem2ko_de, trem2r47h_de, xmg_de, omg_de], ignore_index=True)\n",
    "combiend_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3433</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>51363.949143</td>\n",
       "      <td>7.076655</td>\n",
       "      <td>2.431514e-161</td>\n",
       "      <td>3.661131e-157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3627</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>31139.923458</td>\n",
       "      <td>7.208627</td>\n",
       "      <td>7.782382e-159</td>\n",
       "      <td>5.858966e-155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>91543</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>17487.285409</td>\n",
       "      <td>7.078944</td>\n",
       "      <td>1.216976e-151</td>\n",
       "      <td>6.108001e-148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6373</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>8241.154933</td>\n",
       "      <td>8.971610</td>\n",
       "      <td>2.625696e-140</td>\n",
       "      <td>9.883776e-137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8743</td>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5885.554918</td>\n",
       "      <td>8.436834</td>\n",
       "      <td>4.660559e-135</td>\n",
       "      <td>1.403481e-131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117970</th>\n",
       "      <td>4570</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3017.339959</td>\n",
       "      <td>-1.042412</td>\n",
       "      <td>3.818550e-06</td>\n",
       "      <td>1.901385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117971</th>\n",
       "      <td>4555</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>74.789086</td>\n",
       "      <td>1.586012</td>\n",
       "      <td>4.269662e-11</td>\n",
       "      <td>1.484118e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117972</th>\n",
       "      <td>4509</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5001.407250</td>\n",
       "      <td>1.512440</td>\n",
       "      <td>6.821825e-14</td>\n",
       "      <td>4.403732e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117973</th>\n",
       "      <td>4508</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>35394.848958</td>\n",
       "      <td>1.197526</td>\n",
       "      <td>2.689226e-12</td>\n",
       "      <td>1.429640e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117974</th>\n",
       "      <td>4576</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>118.630934</td>\n",
       "      <td>1.638993</td>\n",
       "      <td>8.873869e-11</td>\n",
       "      <td>2.916276e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117975 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        entrez  condition cell_type      baseMean  log2FoldChange  \\\n",
       "0         3433        IFN      iPSC  51363.949143        7.076655   \n",
       "1         3627        IFN      iPSC  31139.923458        7.208627   \n",
       "2        91543        IFN      iPSC  17487.285409        7.078944   \n",
       "3         6373        IFN      iPSC   8241.154933        8.971610   \n",
       "4         8743        IFN      iPSC   5885.554918        8.436834   \n",
       "...        ...        ...       ...           ...             ...   \n",
       "117970    4570  coculture      iPSC   3017.339959       -1.042412   \n",
       "117971    4555  coculture      iPSC     74.789086        1.586012   \n",
       "117972    4509  coculture      iPSC   5001.407250        1.512440   \n",
       "117973    4508  coculture      iPSC  35394.848958        1.197526   \n",
       "117974    4576  coculture      iPSC    118.630934        1.638993   \n",
       "\n",
       "               pvalue           padj  \n",
       "0       2.431514e-161  3.661131e-157  \n",
       "1       7.782382e-159  5.858966e-155  \n",
       "2       1.216976e-151  6.108001e-148  \n",
       "3       2.625696e-140  9.883776e-137  \n",
       "4       4.660559e-135  1.403481e-131  \n",
       "...               ...            ...  \n",
       "117970   3.818550e-06   1.901385e-04  \n",
       "117971   4.269662e-11   1.484118e-08  \n",
       "117972   6.821825e-14   4.403732e-11  \n",
       "117973   2.689226e-12   1.429640e-09  \n",
       "117974   8.873869e-11   2.916276e-08  \n",
       "\n",
       "[117975 rows x 7 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_de = combiend_de.drop_duplicates(subset=['entrez', 'condition', 'cell_type'])\n",
    "unique_de"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "combiend_de.to_csv('combined_de.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Biological Pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pathways</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SA CASPASE CASCADE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SA FAS SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SA G1 AND S PHASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SA G2 AND M PHASES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3913</th>\n",
       "      <td>WP WNT SIGNALING AND PLURIPOTENCY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3914</th>\n",
       "      <td>WP WNT SIGNALING IN KIDNEY DISEASE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3915</th>\n",
       "      <td>WP WNT SIGNALING WP363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3916</th>\n",
       "      <td>WP WNT SIGNALING WP428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3917</th>\n",
       "      <td>WP ZINC HOMEOSTASIS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3917 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                pathways\n",
       "1           SA B CELL RECEPTOR COMPLEXES\n",
       "2                     SA CASPASE CASCADE\n",
       "3                       SA FAS SIGNALING\n",
       "4                     SA G1 AND S PHASES\n",
       "5                     SA G2 AND M PHASES\n",
       "...                                  ...\n",
       "3913   WP WNT SIGNALING AND PLURIPOTENCY\n",
       "3914  WP WNT SIGNALING IN KIDNEY DISEASE\n",
       "3915              WP WNT SIGNALING WP363\n",
       "3916              WP WNT SIGNALING WP428\n",
       "3917                 WP ZINC HOMEOSTASIS\n",
       "\n",
       "[3917 rows x 1 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathways = pd.read_csv('/Users/nathan/Desktop/Lei\\'s Lab/AD_Microglia_DE_Analysis/AD_Microglia_DEGs/DEA/biological_pathways.csv', index_col=0)\n",
    "pathways['pathways'] = pathways['pathways'].str.replace('_',' ')\n",
    "pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathways.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/pathways.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pathways - Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns in wtc11_ifnb_gsea.csv:\n",
      "['Unnamed: 0', 'pathway', 'pval', 'padj', 'log2err', 'ES', 'NES', 'size', 'gene']\n",
      "----------------------------------------\n",
      "Columns in TREM2_R47H_gsea_genes_c2.csv:\n",
      "['pathway', 'NES', 'padj', 'gene_id']\n",
      "----------------------------------------\n",
      "Columns in pathway_genes_c2_canonical_iMG_vs_xMG7days.csv:\n",
      "['ensembl_gene_id', 'gs_name', 'entrez_gene', 'gene_symbol', 'pathway', 'clean_pathway_name', 'nes', 'padj']\n",
      "----------------------------------------\n",
      "Columns in TREM2_KO_gsea_genes_c2.csv:\n",
      "['pathway', 'NES', 'padj', 'gene_id']\n",
      "----------------------------------------\n",
      "Columns in h1_ifnb_gsea.csv:\n",
      "['Unnamed: 0', 'pathway', 'pval', 'padj', 'log2err', 'ES', 'NES', 'size', 'gene']\n",
      "----------------------------------------\n",
      "Columns in pathway_genes_c2_canonical_iMG_vs_oMG1.csv:\n",
      "['ensembl_gene_id', 'gs_name', 'entrez_gene', 'gene_symbol', 'pathway', 'clean_pathway_name', 'nes', 'padj']\n",
      "----------------------------------------\n"
     ]
    }
   ],
   "source": [
    "gsea_dir = '/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA'\n",
    "csv_files = glob.glob(os.path.join(gsea_dir, '*.csv'))\n",
    "for file in csv_files:\n",
    "    df = pd.read_csv(file)\n",
    "    print(f\"Columns in {os.path.basename(file)}:\")\n",
    "    print(df.columns.tolist())\n",
    "    print('-' * 40) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3669</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>91543</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3433</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3429</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3434</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38082</th>\n",
       "      <td>6171</td>\n",
       "      <td>KEGG MEDICUS REFERENCE TRANSLATION INITIATION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38083</th>\n",
       "      <td>6208</td>\n",
       "      <td>KEGG MEDICUS REFERENCE TRANSLATION INITIATION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38084</th>\n",
       "      <td>6167</td>\n",
       "      <td>KEGG MEDICUS REFERENCE TRANSLATION INITIATION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38085</th>\n",
       "      <td>6187</td>\n",
       "      <td>KEGG MEDICUS REFERENCE TRANSLATION INITIATION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38086</th>\n",
       "      <td>6217</td>\n",
       "      <td>KEGG MEDICUS REFERENCE TRANSLATION INITIATION</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38086 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      entrez                                        pathway\n",
       "1       3669       REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "2      91543       REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "3       3433       REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "4       3429       REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "5       3434       REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "...      ...                                            ...\n",
       "38082   6171  KEGG MEDICUS REFERENCE TRANSLATION INITIATION\n",
       "38083   6208  KEGG MEDICUS REFERENCE TRANSLATION INITIATION\n",
       "38084   6167  KEGG MEDICUS REFERENCE TRANSLATION INITIATION\n",
       "38085   6187  KEGG MEDICUS REFERENCE TRANSLATION INITIATION\n",
       "38086   6217  KEGG MEDICUS REFERENCE TRANSLATION INITIATION\n",
       "\n",
       "[38086 rows x 2 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wtc11_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/wtc11_ifnb_gsea.csv', index_col=0)\n",
    "wtc11_pathways = wtc11_pathways[['gene', 'pathway']]\n",
    "wtc11_pathways['gene'] = wtc11_pathways['gene'].str.replace(\"\\\"\", '')\n",
    "wtc11_pathways['pathway'] = wtc11_pathways['pathway'].str.replace(\"_\", ' ')\n",
    "wtc11_pathways.columns = ['entrez', 'pathway']\n",
    "wtc11_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>91543</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3669</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8638</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3434</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4939</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35540</th>\n",
       "      <td>3990</td>\n",
       "      <td>REACTOME PLASMA LIPOPROTEIN REMODELING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35541</th>\n",
       "      <td>4547</td>\n",
       "      <td>REACTOME PLASMA LIPOPROTEIN REMODELING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35542</th>\n",
       "      <td>3931</td>\n",
       "      <td>REACTOME PLASMA LIPOPROTEIN REMODELING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35543</th>\n",
       "      <td>348</td>\n",
       "      <td>REACTOME PLASMA LIPOPROTEIN REMODELING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35544</th>\n",
       "      <td>55908</td>\n",
       "      <td>REACTOME PLASMA LIPOPROTEIN REMODELING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35545 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      entrez                                   pathway\n",
       "0      91543  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "1       3669  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "2       8638  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "3       3434  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "4       4939  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "...      ...                                       ...\n",
       "35540   3990    REACTOME PLASMA LIPOPROTEIN REMODELING\n",
       "35541   4547    REACTOME PLASMA LIPOPROTEIN REMODELING\n",
       "35542   3931    REACTOME PLASMA LIPOPROTEIN REMODELING\n",
       "35543    348    REACTOME PLASMA LIPOPROTEIN REMODELING\n",
       "35544  55908    REACTOME PLASMA LIPOPROTEIN REMODELING\n",
       "\n",
       "[35545 rows x 2 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h1_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/h1_ifnb_gsea.csv')\n",
    "h1_pathways = h1_pathways[['gene', 'pathway']]\n",
    "h1_pathways['gene'] = h1_pathways['gene'].str.replace(\"\\\"\", '')\n",
    "h1_pathways['pathway'] = h1_pathways['pathway'].str.replace(\"_\", ' ')\n",
    "h1_pathways.columns = ['entrez', 'pathway']\n",
    "h1_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>53632</td>\n",
       "      <td>REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5563</td>\n",
       "      <td>REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6794</td>\n",
       "      <td>REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>389541</td>\n",
       "      <td>REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>64223</td>\n",
       "      <td>REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45698</th>\n",
       "      <td>1305</td>\n",
       "      <td>WP VITAMIN D RECEPTOR PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45699</th>\n",
       "      <td>57167</td>\n",
       "      <td>WP VITAMIN D RECEPTOR PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45700</th>\n",
       "      <td>820</td>\n",
       "      <td>WP VITAMIN D RECEPTOR PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45701</th>\n",
       "      <td>200407</td>\n",
       "      <td>WP VITAMIN D RECEPTOR PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45702</th>\n",
       "      <td>795</td>\n",
       "      <td>WP VITAMIN D RECEPTOR PATHWAY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>45703 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       entrez                                            pathway\n",
       "0       53632  REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...\n",
       "1        5563  REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...\n",
       "2        6794  REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...\n",
       "3      389541  REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...\n",
       "4       64223  REACTOME ENERGY DEPENDENT REGULATION OF MTOR B...\n",
       "...       ...                                                ...\n",
       "45698    1305                      WP VITAMIN D RECEPTOR PATHWAY\n",
       "45699   57167                      WP VITAMIN D RECEPTOR PATHWAY\n",
       "45700     820                      WP VITAMIN D RECEPTOR PATHWAY\n",
       "45701  200407                      WP VITAMIN D RECEPTOR PATHWAY\n",
       "45702     795                      WP VITAMIN D RECEPTOR PATHWAY\n",
       "\n",
       "[45703 rows x 2 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trem2ko_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/TREM2_KO_gsea_genes_c2.csv')\n",
    "trem2ko_pathways = trem2ko_pathways[['gene_id', 'pathway']]\n",
    "trem2ko_pathways['pathway'] = trem2ko_pathways['pathway'].str.replace(\"_\", ' ')\n",
    "trem2ko_pathways.columns = ['entrez', 'pathway']\n",
    "trem2ko_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2167</td>\n",
       "      <td>REACTOME TRIGLYCERIDE METABOLISM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2172</td>\n",
       "      <td>REACTOME TRIGLYCERIDE METABOLISM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>346606</td>\n",
       "      <td>REACTOME TRIGLYCERIDE METABOLISM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>392636</td>\n",
       "      <td>REACTOME TRIGLYCERIDE METABOLISM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5346</td>\n",
       "      <td>REACTOME TRIGLYCERIDE METABOLISM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43152</th>\n",
       "      <td>10581</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43153</th>\n",
       "      <td>3664</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43154</th>\n",
       "      <td>6772</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43155</th>\n",
       "      <td>5610</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43156</th>\n",
       "      <td>10379</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43157 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       entrez                                   pathway\n",
       "0        2167          REACTOME TRIGLYCERIDE METABOLISM\n",
       "1        2172          REACTOME TRIGLYCERIDE METABOLISM\n",
       "2      346606          REACTOME TRIGLYCERIDE METABOLISM\n",
       "3      392636          REACTOME TRIGLYCERIDE METABOLISM\n",
       "4        5346          REACTOME TRIGLYCERIDE METABOLISM\n",
       "...       ...                                       ...\n",
       "43152   10581  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "43153    3664  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "43154    6772  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "43155    5610  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "43156   10379  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "\n",
       "[43157 rows x 2 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trem2r47h_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/TREM2_R47H_gsea_genes_c2.csv')\n",
    "trem2r47h_pathways = trem2r47h_pathways[['gene_id', 'pathway']]\n",
    "trem2r47h_pathways['pathway'] = trem2r47h_pathways['pathway'].str.replace(\"_\", ' ')\n",
    "trem2r47h_pathways.columns = ['entrez', 'pathway']\n",
    "trem2r47h_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1636</td>\n",
       "      <td>SA MMP CYTOKINE CONNECTION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>960</td>\n",
       "      <td>SA MMP CYTOKINE CONNECTION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2214</td>\n",
       "      <td>SA MMP CYTOKINE CONNECTION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>944</td>\n",
       "      <td>SA MMP CYTOKINE CONNECTION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2318</td>\n",
       "      <td>SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>23533</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>23368</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>57580</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>6295</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>6295</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     entrez                                            pathway\n",
       "0      1636                         SA MMP CYTOKINE CONNECTION\n",
       "1       960                         SA MMP CYTOKINE CONNECTION\n",
       "2      2214                         SA MMP CYTOKINE CONNECTION\n",
       "3       944                         SA MMP CYTOKINE CONNECTION\n",
       "4      2318  SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...\n",
       "..      ...                                                ...\n",
       "105   23533                SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "106   23368                SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "107   57580                SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "108    6295                SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "109    6295                SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "\n",
       "[110 rows x 2 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "xenot7d_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/pathway_genes_c2_canonical_iMG_vs_xMG7days.csv')\n",
    "xenot7d_pathways = xenot7d_pathways[['entrez_gene', 'clean_pathway_name']]\n",
    "xenot7d_pathways.columns = ['entrez', 'pathway']\n",
    "xenot7d_pathways\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85366</td>\n",
       "      <td>SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5063</td>\n",
       "      <td>SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>57144</td>\n",
       "      <td>SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5600</td>\n",
       "      <td>SIG CD40PATHWAYMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5603</td>\n",
       "      <td>SIG CD40PATHWAYMAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>4067</td>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>5604</td>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>6197</td>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>7409</td>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>10451</td>\n",
       "      <td>SA B CELL RECEPTOR COMPLEXES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     entrez                                            pathway\n",
       "0     85366  SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...\n",
       "1      5063  SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...\n",
       "2     57144  SIG REGULATION OF THE ACTIN CYTOSKELETON BY RH...\n",
       "3      5600                                 SIG CD40PATHWAYMAP\n",
       "4      5603                                 SIG CD40PATHWAYMAP\n",
       "..      ...                                                ...\n",
       "115    4067                       SA B CELL RECEPTOR COMPLEXES\n",
       "116    5604                       SA B CELL RECEPTOR COMPLEXES\n",
       "117    6197                       SA B CELL RECEPTOR COMPLEXES\n",
       "118    7409                       SA B CELL RECEPTOR COMPLEXES\n",
       "119   10451                       SA B CELL RECEPTOR COMPLEXES\n",
       "\n",
       "[120 rows x 2 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coculture_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/GSEA/pathway_genes_c2_canonical_iMG_vs_oMG1.csv')\n",
    "coculture_pathways = coculture_pathways[['entrez_gene', 'clean_pathway_name']]\n",
    "coculture_pathways.columns = ['entrez', 'pathway']\n",
    "coculture_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "114211\n",
      "114211\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3669</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>91543</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3433</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3429</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3434</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162655</th>\n",
       "      <td>5336</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162663</th>\n",
       "      <td>7409</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162692</th>\n",
       "      <td>207</td>\n",
       "      <td>SA PTEN PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162697</th>\n",
       "      <td>2885</td>\n",
       "      <td>SA PTEN PATHWAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162703</th>\n",
       "      <td>5290</td>\n",
       "      <td>SA PTEN PATHWAY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114211 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       entrez                                   pathway\n",
       "0        3669  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "1       91543  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "2        3433  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "3        3429  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "4        3434  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "...       ...                                       ...\n",
       "162655   5336       SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "162663   7409       SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "162692    207                           SA PTEN PATHWAY\n",
       "162697   2885                           SA PTEN PATHWAY\n",
       "162703   5290                           SA PTEN PATHWAY\n",
       "\n",
       "[114211 rows x 2 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_gene_pathways = pd.concat([wtc11_pathways, h1_pathways, trem2ko_pathways, trem2r47h_pathways, xenot7d_pathways, coculture_pathways], ignore_index=True)\n",
    "combined_gene_pathways = combined_gene_pathways.drop_duplicates(subset=['entrez', 'pathway'])\n",
    "print(len(combined_gene_pathways))\n",
    "combined_gene_pathways = combined_gene_pathways.dropna()\n",
    "print(len(combined_gene_pathways))\n",
    "combined_gene_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_gene_pathways.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/combined_gene_pathways.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CREs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory = \"./Data/CREs\"\n",
    "for file in os.listdir(directory):\n",
    "    if file.endswith(\".txt\"):\n",
    "        filepath = os.path.join(directory, file)\n",
    "        df = pd.read_csv(filepath, sep='\\t')\n",
    "        df = df.iloc[:, 0:6]\n",
    "        df.columns = [\"unknown\", \"RPM_condition\", \"RPM_control\", \"chr\", \"start\", \"end\"]\n",
    "        df['CRElog2FC'] = np.log2((df[\"RPM_condition\"] + 1) / (df[\"RPM_control\"] + 1))\n",
    "        df.to_csv(filepath, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in os.listdir(directory):\n",
    "    if file.endswith(\".txt\"):\n",
    "        filepath = os.path.join(directory, file)\n",
    "        df = pd.read_csv(filepath, sep='\\t')\n",
    "        df = df.iloc[:, 3:7]\n",
    "        df.to_csv(filepath, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read mapped cres in and perform metadata changes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory = \"./Data/CREs\"\n",
    "outdir1 = \"./Data/Filtered_cres\"\n",
    "outdir2 = \"./Data/Mapped_cres\"\n",
    "for file in os.listdir(directory):\n",
    "    if file.endswith(\".tsv\"):\n",
    "        filepath = os.path.join(directory, file)\n",
    "        df = pd.read_csv(filepath, sep='\\t')\n",
    "        df.columns = [\"chr\", \"start\", \"end\", \"cre_log2foldchange\", \"merged_chr\", \"merged_start\", \"merged_end\", \"tss_chr\", \"tss_start\", \"tss_end\", \"strand\", \"hgnc\", \"entrez\", \"na1\", \"na2\", \"distance_to_tss\"]\n",
    "        if file == \"h1_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'IFN')\n",
    "            df.insert(1, 'cell_type', 'ESC')\n",
    "        elif file == \"wtc11_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'IFN')\n",
    "            df.insert(1, 'cell_type', 'iPSC')\n",
    "        elif file == \"trem2ko_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'TREM2KO')\n",
    "            df.insert(1, 'cell_type', 'ESC')\n",
    "        elif file == \"trem2r47h_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'TREM2R47H')\n",
    "            df.insert(1, 'cell_type', 'ESC')\n",
    "        elif file == \"xenot7d_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'xenot7d')\n",
    "            df.insert(1, 'cell_type', 'iPSC')\n",
    "        elif file == \"coculture_mapped_cres.tsv\":\n",
    "            df.insert(0, 'condition', 'coculture')\n",
    "            df.insert(1, 'cell_type', 'iPSC')\n",
    "        cres = df[['condition', 'cell_type', 'chr', 'start', 'end', 'cre_log2foldchange', \"merged_chr\", \"merged_start\", \"merged_end\"]]\n",
    "        cres_mapped = df[['condition', 'cell_type', 'chr', 'start', 'end', \"entrez\", \"distance_to_tss\"]]\n",
    "        outpath1 = os.path.join(outdir1, file)\n",
    "        outpath2 = os.path.join(outdir2, file)\n",
    "        cres.to_csv(outpath1, sep='\\t', index=False)\n",
    "        cres_mapped.to_csv(outpath2, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read in filtered cres, combine them into one table and drop duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "h1_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/h1_mapped_cres.tsv\", sep=\"\\t\")\n",
    "wtc11_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/wtc11_mapped_cres.tsv\", sep=\"\\t\")\n",
    "trem2ko_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/trem2ko_mapped_cres.tsv\", sep=\"\\t\")\n",
    "trem2r47h_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/trem2r47h_mapped_cres.tsv\", sep=\"\\t\")\n",
    "xenot7d_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/xenot7d_mapped_cres.tsv\", sep=\"\\t\")\n",
    "coculture_cres = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Filtered_cres/coculture_mapped_cres.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>cre_log2foldchange</th>\n",
       "      <th>merged_chr</th>\n",
       "      <th>merged_start</th>\n",
       "      <th>merged_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89295608</td>\n",
       "      <td>89296790</td>\n",
       "      <td>0.434959</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89295506</td>\n",
       "      <td>89297078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89301577</td>\n",
       "      <td>89302330</td>\n",
       "      <td>-0.104734</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89301099</td>\n",
       "      <td>89303218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "      <td>2.769050</td>\n",
       "      <td>chr10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>chr2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>0.597089</td>\n",
       "      <td>chr2</td>\n",
       "      <td>6863605</td>\n",
       "      <td>6867024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>chr2</td>\n",
       "      <td>6877401</td>\n",
       "      <td>6878097</td>\n",
       "      <td>2.152570</td>\n",
       "      <td>chr2</td>\n",
       "      <td>6875992</td>\n",
       "      <td>6878409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361158</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>chrX</td>\n",
       "      <td>11996722</td>\n",
       "      <td>11997282</td>\n",
       "      <td>-0.299677</td>\n",
       "      <td>chrX</td>\n",
       "      <td>11996623</td>\n",
       "      <td>11997551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361159</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12301160</td>\n",
       "      <td>12301819</td>\n",
       "      <td>-1.798817</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12301069</td>\n",
       "      <td>12301966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361160</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12498107</td>\n",
       "      <td>12498700</td>\n",
       "      <td>-1.062652</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12497991</td>\n",
       "      <td>12498808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361161</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12506691</td>\n",
       "      <td>12507175</td>\n",
       "      <td>-0.404536</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12506606</td>\n",
       "      <td>12507175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361162</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12515307</td>\n",
       "      <td>12515876</td>\n",
       "      <td>-2.047743</td>\n",
       "      <td>chrX</td>\n",
       "      <td>12515127</td>\n",
       "      <td>12516039</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>361163 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type    chr     start       end  cre_log2foldchange  \\\n",
       "0             IFN       ESC  chr10  89295608  89296790            0.434959   \n",
       "1             IFN       ESC  chr10  89301577  89302330           -0.104734   \n",
       "2             IFN       ESC  chr10  89309356  89310733            2.769050   \n",
       "3             IFN       ESC   chr2   6865106   6866522            0.597089   \n",
       "4             IFN       ESC   chr2   6877401   6878097            2.152570   \n",
       "...           ...       ...    ...       ...       ...                 ...   \n",
       "361158  coculture      iPSC   chrX  11996722  11997282           -0.299677   \n",
       "361159  coculture      iPSC   chrX  12301160  12301819           -1.798817   \n",
       "361160  coculture      iPSC   chrX  12498107  12498700           -1.062652   \n",
       "361161  coculture      iPSC   chrX  12506691  12507175           -0.404536   \n",
       "361162  coculture      iPSC   chrX  12515307  12515876           -2.047743   \n",
       "\n",
       "       merged_chr  merged_start  merged_end  \n",
       "0           chr10      89295506    89297078  \n",
       "1           chr10      89301099    89303218  \n",
       "2           chr10      89309356    89310733  \n",
       "3            chr2       6863605     6867024  \n",
       "4            chr2       6875992     6878409  \n",
       "...           ...           ...         ...  \n",
       "361158       chrX      11996623    11997551  \n",
       "361159       chrX      12301069    12301966  \n",
       "361160       chrX      12497991    12498808  \n",
       "361161       chrX      12506606    12507175  \n",
       "361162       chrX      12515127    12516039  \n",
       "\n",
       "[361163 rows x 9 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cres = pd.concat([h1_cres, wtc11_cres, trem2ko_cres, trem2r47h_cres, xenot7d_cres, coculture_cres], ignore_index=True)\n",
    "cres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y_/tfywlws16k71l2k62r0j683m0000gn/T/ipykernel_95578/501924833.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  unique_cres['chr'] = unique_cres['chr'].str.replace('chr', '')\n",
      "/var/folders/y_/tfywlws16k71l2k62r0j683m0000gn/T/ipykernel_95578/501924833.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  unique_cres['merged_chr'] = unique_cres['merged_chr'].str.replace('chr', '')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>cre_log2foldchange</th>\n",
       "      <th>merged_chr</th>\n",
       "      <th>merged_start</th>\n",
       "      <th>merged_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89295608</td>\n",
       "      <td>89296790</td>\n",
       "      <td>0.434959</td>\n",
       "      <td>10</td>\n",
       "      <td>89295506</td>\n",
       "      <td>89297078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89301577</td>\n",
       "      <td>89302330</td>\n",
       "      <td>-0.104734</td>\n",
       "      <td>10</td>\n",
       "      <td>89301099</td>\n",
       "      <td>89303218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "      <td>2.769050</td>\n",
       "      <td>10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>0.597089</td>\n",
       "      <td>2</td>\n",
       "      <td>6863605</td>\n",
       "      <td>6867024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6877401</td>\n",
       "      <td>6878097</td>\n",
       "      <td>2.152570</td>\n",
       "      <td>2</td>\n",
       "      <td>6875992</td>\n",
       "      <td>6878409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361158</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>11996722</td>\n",
       "      <td>11997282</td>\n",
       "      <td>-0.299677</td>\n",
       "      <td>X</td>\n",
       "      <td>11996623</td>\n",
       "      <td>11997551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361159</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12301160</td>\n",
       "      <td>12301819</td>\n",
       "      <td>-1.798817</td>\n",
       "      <td>X</td>\n",
       "      <td>12301069</td>\n",
       "      <td>12301966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361160</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12498107</td>\n",
       "      <td>12498700</td>\n",
       "      <td>-1.062652</td>\n",
       "      <td>X</td>\n",
       "      <td>12497991</td>\n",
       "      <td>12498808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361161</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12506691</td>\n",
       "      <td>12507175</td>\n",
       "      <td>-0.404536</td>\n",
       "      <td>X</td>\n",
       "      <td>12506606</td>\n",
       "      <td>12507175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361162</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12515307</td>\n",
       "      <td>12515876</td>\n",
       "      <td>-2.047743</td>\n",
       "      <td>X</td>\n",
       "      <td>12515127</td>\n",
       "      <td>12516039</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340983 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr     start       end  cre_log2foldchange  \\\n",
       "0             IFN       ESC  10  89295608  89296790            0.434959   \n",
       "1             IFN       ESC  10  89301577  89302330           -0.104734   \n",
       "2             IFN       ESC  10  89309356  89310733            2.769050   \n",
       "3             IFN       ESC   2   6865106   6866522            0.597089   \n",
       "4             IFN       ESC   2   6877401   6878097            2.152570   \n",
       "...           ...       ...  ..       ...       ...                 ...   \n",
       "361158  coculture      iPSC   X  11996722  11997282           -0.299677   \n",
       "361159  coculture      iPSC   X  12301160  12301819           -1.798817   \n",
       "361160  coculture      iPSC   X  12498107  12498700           -1.062652   \n",
       "361161  coculture      iPSC   X  12506691  12507175           -0.404536   \n",
       "361162  coculture      iPSC   X  12515307  12515876           -2.047743   \n",
       "\n",
       "       merged_chr  merged_start  merged_end  \n",
       "0              10      89295506    89297078  \n",
       "1              10      89301099    89303218  \n",
       "2              10      89309356    89310733  \n",
       "3               2       6863605     6867024  \n",
       "4               2       6875992     6878409  \n",
       "...           ...           ...         ...  \n",
       "361158          X      11996623    11997551  \n",
       "361159          X      12301069    12301966  \n",
       "361160          X      12497991    12498808  \n",
       "361161          X      12506606    12507175  \n",
       "361162          X      12515127    12516039  \n",
       "\n",
       "[340983 rows x 9 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_cres = cres.drop_duplicates(subset=['condition', 'cell_type', 'chr', 'start', 'end'])\n",
    "unique_cres['chr'] = unique_cres['chr'].str.replace('chr', '')\n",
    "unique_cres['merged_chr'] = unique_cres['merged_chr'].str.replace('chr', '')\n",
    "unique_cres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_cres.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/unique_cres.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "now for cres linked to genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "h1_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/h1_mapped_cres.tsv\", sep=\"\\t\")\n",
    "wtc11_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/wtc11_mapped_cres.tsv\", sep=\"\\t\")\n",
    "trem2ko_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/trem2ko_mapped_cres.tsv\", sep=\"\\t\")\n",
    "trem2r47h_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/trem2r47h_mapped_cres.tsv\", sep=\"\\t\")\n",
    "xenot7d_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/xenot7d_mapped_cres.tsv\", sep=\"\\t\")\n",
    "coculture_cres_genes = pd.read_csv(\"/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Data/Mapped_cres/coculture_mapped_cres.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "361163\n",
      "360526\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entrez</th>\n",
       "      <th>distance_to_tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89295608</td>\n",
       "      <td>89296790</td>\n",
       "      <td>3433</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89301577</td>\n",
       "      <td>89302330</td>\n",
       "      <td>3433</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "      <td>3433</td>\n",
       "      <td>3395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>91543</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6877401</td>\n",
       "      <td>6878097</td>\n",
       "      <td>91543</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361158</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>11996722</td>\n",
       "      <td>11997282</td>\n",
       "      <td>9758</td>\n",
       "      <td>118862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361159</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12301160</td>\n",
       "      <td>12301819</td>\n",
       "      <td>9758</td>\n",
       "      <td>39726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361160</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12498107</td>\n",
       "      <td>12498700</td>\n",
       "      <td>9758</td>\n",
       "      <td>3143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361161</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12506691</td>\n",
       "      <td>12507175</td>\n",
       "      <td>9758</td>\n",
       "      <td>11727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361162</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12515307</td>\n",
       "      <td>12515876</td>\n",
       "      <td>9758</td>\n",
       "      <td>20343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>360526 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr     start       end  entrez  distance_to_tss\n",
       "0             IFN       ESC  10  89295608  89296790    3433             3087\n",
       "1             IFN       ESC  10  89301577  89302330    3433                0\n",
       "2             IFN       ESC  10  89309356  89310733    3433             3395\n",
       "3             IFN       ESC   2   6865106   6866522   91543                0\n",
       "4             IFN       ESC   2   6877401   6878097   91543                0\n",
       "...           ...       ...  ..       ...       ...     ...              ...\n",
       "361158  coculture      iPSC   X  11996722  11997282    9758           118862\n",
       "361159  coculture      iPSC   X  12301160  12301819    9758            39726\n",
       "361160  coculture      iPSC   X  12498107  12498700    9758             3143\n",
       "361161  coculture      iPSC   X  12506691  12507175    9758            11727\n",
       "361162  coculture      iPSC   X  12515307  12515876    9758            20343\n",
       "\n",
       "[360526 rows x 7 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_cres_genes = pd.concat([h1_cres_genes, wtc11_cres_genes, trem2ko_cres_genes, trem2r47h_cres_genes, xenot7d_cres_genes, coculture_cres_genes], ignore_index=True)\n",
    "print(len(unique_cres_genes))\n",
    "unique_cres_genes = unique_cres_genes.drop_duplicates(subset=['condition', 'cell_type','chr', 'start', 'end', 'entrez'])\n",
    "print(len(unique_cres_genes))\n",
    "unique_cres_genes['chr'] = unique_cres_genes['chr'].str.replace('chr', '')\n",
    "unique_cres_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicates = unique_cres_genes.groupby(['start', 'end']) \\\n",
    "                              .filter(lambda x: len(x) > 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entrez</th>\n",
       "      <th>distance_to_tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>91543</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>3</td>\n",
       "      <td>57227304</td>\n",
       "      <td>57228821</td>\n",
       "      <td>8820</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>129607</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>162318011</td>\n",
       "      <td>162319366</td>\n",
       "      <td>64135</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>6</td>\n",
       "      <td>32853316</td>\n",
       "      <td>32854532</td>\n",
       "      <td>5698</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361158</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>11996722</td>\n",
       "      <td>11997282</td>\n",
       "      <td>9758</td>\n",
       "      <td>118862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361159</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12301160</td>\n",
       "      <td>12301819</td>\n",
       "      <td>9758</td>\n",
       "      <td>39726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361160</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12498107</td>\n",
       "      <td>12498700</td>\n",
       "      <td>9758</td>\n",
       "      <td>3143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361161</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12506691</td>\n",
       "      <td>12507175</td>\n",
       "      <td>9758</td>\n",
       "      <td>11727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361162</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12515307</td>\n",
       "      <td>12515876</td>\n",
       "      <td>9758</td>\n",
       "      <td>20343</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>305467 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr      start        end  entrez  distance_to_tss\n",
       "3             IFN       ESC   2    6865106    6866522   91543                0\n",
       "14            IFN       ESC   3   57227304   57228821    8820                0\n",
       "15            IFN       ESC   2    6865106    6866522  129607                0\n",
       "39            IFN       ESC   2  162318011  162319366   64135                0\n",
       "42            IFN       ESC   6   32853316   32854532    5698                0\n",
       "...           ...       ...  ..        ...        ...     ...              ...\n",
       "361158  coculture      iPSC   X   11996722   11997282    9758           118862\n",
       "361159  coculture      iPSC   X   12301160   12301819    9758            39726\n",
       "361160  coculture      iPSC   X   12498107   12498700    9758             3143\n",
       "361161  coculture      iPSC   X   12506691   12507175    9758            11727\n",
       "361162  coculture      iPSC   X   12515307   12515876    9758            20343\n",
       "\n",
       "[305467 rows x 7 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_cres_genes.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/unique_cres_genes.csv', index=False) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final Checks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31020\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFIT2</td>\n",
       "      <td>ENSG00000119922</td>\n",
       "      <td>3433</td>\n",
       "      <td>10</td>\n",
       "      <td>89283694.0</td>\n",
       "      <td>89309271.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CXCL10</td>\n",
       "      <td>ENSG00000169245</td>\n",
       "      <td>3627</td>\n",
       "      <td>4</td>\n",
       "      <td>76021118.0</td>\n",
       "      <td>76023497.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RSAD2</td>\n",
       "      <td>ENSG00000134321</td>\n",
       "      <td>91543</td>\n",
       "      <td>2</td>\n",
       "      <td>6865557.0</td>\n",
       "      <td>6898239.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXCL11</td>\n",
       "      <td>ENSG00000169248</td>\n",
       "      <td>6373</td>\n",
       "      <td>4</td>\n",
       "      <td>76033682.0</td>\n",
       "      <td>76041415.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TNFSF10</td>\n",
       "      <td>ENSG00000121858</td>\n",
       "      <td>8743</td>\n",
       "      <td>3</td>\n",
       "      <td>172505508.0</td>\n",
       "      <td>172523475.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31015</th>\n",
       "      <td>PAGE4</td>\n",
       "      <td>ENSG00000101951</td>\n",
       "      <td>9506</td>\n",
       "      <td>X</td>\n",
       "      <td>49793653.0</td>\n",
       "      <td>49834264.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31016</th>\n",
       "      <td>GAL3ST1</td>\n",
       "      <td>ENSG00000128242</td>\n",
       "      <td>9514</td>\n",
       "      <td>22</td>\n",
       "      <td>30554635.0</td>\n",
       "      <td>30574665.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31017</th>\n",
       "      <td>CD40LG</td>\n",
       "      <td>ENSG00000102245</td>\n",
       "      <td>959</td>\n",
       "      <td>X</td>\n",
       "      <td>136648158.0</td>\n",
       "      <td>136660390.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31018</th>\n",
       "      <td>FRMPD4</td>\n",
       "      <td>ENSG00000169933</td>\n",
       "      <td>9758</td>\n",
       "      <td>X</td>\n",
       "      <td>11822423.0</td>\n",
       "      <td>12724523.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31019</th>\n",
       "      <td>MAGEC1</td>\n",
       "      <td>ENSG00000155495</td>\n",
       "      <td>9947</td>\n",
       "      <td>X</td>\n",
       "      <td>141903894.0</td>\n",
       "      <td>141909374.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31020 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          hgnc  ensembl_gene_id  entrez chromosome  start_position  \\\n",
       "0        IFIT2  ENSG00000119922    3433         10      89283694.0   \n",
       "1       CXCL10  ENSG00000169245    3627          4      76021118.0   \n",
       "2        RSAD2  ENSG00000134321   91543          2       6865557.0   \n",
       "3       CXCL11  ENSG00000169248    6373          4      76033682.0   \n",
       "4      TNFSF10  ENSG00000121858    8743          3     172505508.0   \n",
       "...        ...              ...     ...        ...             ...   \n",
       "31015    PAGE4  ENSG00000101951    9506          X      49793653.0   \n",
       "31016  GAL3ST1  ENSG00000128242    9514         22      30554635.0   \n",
       "31017   CD40LG  ENSG00000102245     959          X     136648158.0   \n",
       "31018   FRMPD4  ENSG00000169933    9758          X      11822423.0   \n",
       "31019   MAGEC1  ENSG00000155495    9947          X     141903894.0   \n",
       "\n",
       "       end_position strand  \n",
       "0        89309271.0      +  \n",
       "1        76023497.0      -  \n",
       "2         6898239.0      +  \n",
       "3        76041415.0      -  \n",
       "4       172523475.0      -  \n",
       "...             ...    ...  \n",
       "31015    49834264.0      +  \n",
       "31016    30574665.0      -  \n",
       "31017   136660390.0      +  \n",
       "31018    12724523.0      +  \n",
       "31019   141909374.0      +  \n",
       "\n",
       "[31020 rows x 7 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_genes = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/combined_genes.csv')\n",
    "print(len(final_genes))\n",
    "#final_genes = final_genes[final_genes['entrez'].isna()]\n",
    "final_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "117975\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>entrez</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [hgnc, ensembl_gene_id, entrez, chromosome, start_position, end_position, strand, condition, cell_type, baseMean, log2FoldChange, pvalue, padj]\n",
       "Index: []"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_cell_de = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/combined_de.csv')\n",
    "print(len(final_cell_de))\n",
    "final_cell_de = final_cell_de.drop_duplicates(subset=['entrez', 'condition', 'cell_type'])\n",
    "cross1 = pd.merge(final_genes, final_cell_de, on='entrez', how='left')\n",
    "cross1_na = cross1[cross1['pvalue'].isna()]\n",
    "cross1_na   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "340983\n",
      "340983\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>cre_log2foldchange</th>\n",
       "      <th>merged_chr</th>\n",
       "      <th>merged_start</th>\n",
       "      <th>merged_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89295608</td>\n",
       "      <td>89296790</td>\n",
       "      <td>0.434959</td>\n",
       "      <td>10</td>\n",
       "      <td>89295506</td>\n",
       "      <td>89297078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89301577</td>\n",
       "      <td>89302330</td>\n",
       "      <td>-0.104734</td>\n",
       "      <td>10</td>\n",
       "      <td>89301099</td>\n",
       "      <td>89303218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "      <td>2.769050</td>\n",
       "      <td>10</td>\n",
       "      <td>89309356</td>\n",
       "      <td>89310733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6865106</td>\n",
       "      <td>6866522</td>\n",
       "      <td>0.597089</td>\n",
       "      <td>2</td>\n",
       "      <td>6863605</td>\n",
       "      <td>6867024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>6877401</td>\n",
       "      <td>6878097</td>\n",
       "      <td>2.152570</td>\n",
       "      <td>2</td>\n",
       "      <td>6875992</td>\n",
       "      <td>6878409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340978</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>11996722</td>\n",
       "      <td>11997282</td>\n",
       "      <td>-0.299677</td>\n",
       "      <td>X</td>\n",
       "      <td>11996623</td>\n",
       "      <td>11997551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340979</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12301160</td>\n",
       "      <td>12301819</td>\n",
       "      <td>-1.798817</td>\n",
       "      <td>X</td>\n",
       "      <td>12301069</td>\n",
       "      <td>12301966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340980</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12498107</td>\n",
       "      <td>12498700</td>\n",
       "      <td>-1.062652</td>\n",
       "      <td>X</td>\n",
       "      <td>12497991</td>\n",
       "      <td>12498808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340981</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12506691</td>\n",
       "      <td>12507175</td>\n",
       "      <td>-0.404536</td>\n",
       "      <td>X</td>\n",
       "      <td>12506606</td>\n",
       "      <td>12507175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340982</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>X</td>\n",
       "      <td>12515307</td>\n",
       "      <td>12515876</td>\n",
       "      <td>-2.047743</td>\n",
       "      <td>X</td>\n",
       "      <td>12515127</td>\n",
       "      <td>12516039</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>340983 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr     start       end  cre_log2foldchange  \\\n",
       "0             IFN       ESC  10  89295608  89296790            0.434959   \n",
       "1             IFN       ESC  10  89301577  89302330           -0.104734   \n",
       "2             IFN       ESC  10  89309356  89310733            2.769050   \n",
       "3             IFN       ESC   2   6865106   6866522            0.597089   \n",
       "4             IFN       ESC   2   6877401   6878097            2.152570   \n",
       "...           ...       ...  ..       ...       ...                 ...   \n",
       "340978  coculture      iPSC   X  11996722  11997282           -0.299677   \n",
       "340979  coculture      iPSC   X  12301160  12301819           -1.798817   \n",
       "340980  coculture      iPSC   X  12498107  12498700           -1.062652   \n",
       "340981  coculture      iPSC   X  12506691  12507175           -0.404536   \n",
       "340982  coculture      iPSC   X  12515307  12515876           -2.047743   \n",
       "\n",
       "       merged_chr  merged_start  merged_end  \n",
       "0              10      89295506    89297078  \n",
       "1              10      89301099    89303218  \n",
       "2              10      89309356    89310733  \n",
       "3               2       6863605     6867024  \n",
       "4               2       6875992     6878409  \n",
       "...           ...           ...         ...  \n",
       "340978          X      11996623    11997551  \n",
       "340979          X      12301069    12301966  \n",
       "340980          X      12497991    12498808  \n",
       "340981          X      12506606    12507175  \n",
       "340982          X      12515127    12516039  \n",
       "\n",
       "[340983 rows x 9 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_cres = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/unique_cres.csv')\n",
    "print(len(final_cres))\n",
    "final_cres = final_cres.drop_duplicates(subset=['condition', 'cell_type','chr', 'start', 'end'])\n",
    "print(len(final_cres))\n",
    "#final_cres = final_cres[final_cres[['chr','start', 'end']].isna().any(axis=1)]\n",
    "final_cres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "200049\n",
      "200049\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>797406</td>\n",
       "      <td>798178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>816037</td>\n",
       "      <td>818244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>819296</td>\n",
       "      <td>820336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>842686</td>\n",
       "      <td>843233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200044</th>\n",
       "      <td>Y</td>\n",
       "      <td>21421806</td>\n",
       "      <td>21422749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200045</th>\n",
       "      <td>Y</td>\n",
       "      <td>21440434</td>\n",
       "      <td>21441017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200046</th>\n",
       "      <td>Y</td>\n",
       "      <td>21481413</td>\n",
       "      <td>21482088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200047</th>\n",
       "      <td>Y</td>\n",
       "      <td>26408715</td>\n",
       "      <td>26409392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200048</th>\n",
       "      <td>Y</td>\n",
       "      <td>26476761</td>\n",
       "      <td>26477438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200049 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr     start       end\n",
       "0        1    797406    798178\n",
       "1        1    816037    818244\n",
       "2        1    819296    820336\n",
       "3        1    826435    828307\n",
       "4        1    842686    843233\n",
       "...     ..       ...       ...\n",
       "200044   Y  21421806  21422749\n",
       "200045   Y  21440434  21441017\n",
       "200046   Y  21481413  21482088\n",
       "200047   Y  26408715  26409392\n",
       "200048   Y  26476761  26477438\n",
       "\n",
       "[200049 rows x 3 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_merged = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cres.csv')\n",
    "print(len(final_merged))\n",
    "final_merged = final_merged.drop_duplicates(subset=['chr', 'start', 'end'])\n",
    "print(len(final_merged))\n",
    "final_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "360526\n",
      "360526\n",
      "360526\n",
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "final_cre_gene = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/unique_cres_genes.csv')\n",
    "print(len(final_cre_gene))\n",
    "final_cre_gene = final_cre_gene[final_cre_gene[['chr','start', 'end']].notna().any(axis=1)]\n",
    "print(len(final_cre_gene))\n",
    "final_cre_gene = final_cre_gene.drop_duplicates(subset=['condition', 'cell_type','chr', 'start', 'end','entrez'])\n",
    "print(len(final_cre_gene))\n",
    "cross2 = pd.merge(final_cre_gene, final_cres, on=['condition', 'cell_type','chr', 'start', 'end'], how='inner')\n",
    "cross2 = cross2[cross2['cre_log2foldchange'].isna()]\n",
    "print(len(cross2))\n",
    "cross3 = pd.merge(final_cre_gene, final_genes, on=['entrez'], how='left')\n",
    "cross3 = cross3[cross3['strand'].isna()]\n",
    "print(len(cross3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entrez</th>\n",
       "      <th>distance_to_tss</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23604</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>63124152</td>\n",
       "      <td>63124596</td>\n",
       "      <td>100169989</td>\n",
       "      <td>4611</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000293077</td>\n",
       "      <td>2</td>\n",
       "      <td>63117851.0</td>\n",
       "      <td>63119542.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23614</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>7430624</td>\n",
       "      <td>7431506</td>\n",
       "      <td>100506274</td>\n",
       "      <td>620</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000229727</td>\n",
       "      <td>2</td>\n",
       "      <td>7293688.0</td>\n",
       "      <td>7450544.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23650</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>60759216</td>\n",
       "      <td>60759812</td>\n",
       "      <td>101926964</td>\n",
       "      <td>65772</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000231252</td>\n",
       "      <td>1</td>\n",
       "      <td>60659631.0</td>\n",
       "      <td>60868041.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23651</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>60866777</td>\n",
       "      <td>60867731</td>\n",
       "      <td>101926964</td>\n",
       "      <td>235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000231252</td>\n",
       "      <td>1</td>\n",
       "      <td>60659631.0</td>\n",
       "      <td>60868041.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23652</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>60874520</td>\n",
       "      <td>60875359</td>\n",
       "      <td>101926964</td>\n",
       "      <td>6512</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000231252</td>\n",
       "      <td>1</td>\n",
       "      <td>60659631.0</td>\n",
       "      <td>60868041.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360156</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>21</td>\n",
       "      <td>9839168</td>\n",
       "      <td>9839451</td>\n",
       "      <td>441058</td>\n",
       "      <td>1158</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000293035</td>\n",
       "      <td>21</td>\n",
       "      <td>9781848.0</td>\n",
       "      <td>9839362.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360346</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>22</td>\n",
       "      <td>22558948</td>\n",
       "      <td>22559646</td>\n",
       "      <td>648691</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000220891</td>\n",
       "      <td>22</td>\n",
       "      <td>22559343.0</td>\n",
       "      <td>22566602.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360409</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>22</td>\n",
       "      <td>45996358</td>\n",
       "      <td>45998234</td>\n",
       "      <td>730668</td>\n",
       "      <td>8382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000280424</td>\n",
       "      <td>22</td>\n",
       "      <td>46006616.0</td>\n",
       "      <td>46010777.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360410</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>22</td>\n",
       "      <td>46000867</td>\n",
       "      <td>46001379</td>\n",
       "      <td>730668</td>\n",
       "      <td>5237</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000280424</td>\n",
       "      <td>22</td>\n",
       "      <td>46006616.0</td>\n",
       "      <td>46010777.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360411</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>22</td>\n",
       "      <td>46006234</td>\n",
       "      <td>46007565</td>\n",
       "      <td>730668</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000280424</td>\n",
       "      <td>22</td>\n",
       "      <td>46006616.0</td>\n",
       "      <td>46010777.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5084 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr     start       end     entrez  \\\n",
       "23604         IFN       ESC   2  63124152  63124596  100169989   \n",
       "23614         IFN       ESC   2   7430624   7431506  100506274   \n",
       "23650         IFN       ESC   1  60759216  60759812  101926964   \n",
       "23651         IFN       ESC   1  60866777  60867731  101926964   \n",
       "23652         IFN       ESC   1  60874520  60875359  101926964   \n",
       "...           ...       ...  ..       ...       ...        ...   \n",
       "360156  coculture      iPSC  21   9839168   9839451     441058   \n",
       "360346  coculture      iPSC  22  22558948  22559646     648691   \n",
       "360409  coculture      iPSC  22  45996358  45998234     730668   \n",
       "360410  coculture      iPSC  22  46000867  46001379     730668   \n",
       "360411  coculture      iPSC  22  46006234  46007565     730668   \n",
       "\n",
       "        distance_to_tss hgnc  ensembl_gene_id chromosome  start_position  \\\n",
       "23604              4611  NaN  ENSG00000293077          2      63117851.0   \n",
       "23614               620  NaN  ENSG00000229727          2       7293688.0   \n",
       "23650             65772  NaN  ENSG00000231252          1      60659631.0   \n",
       "23651               235  NaN  ENSG00000231252          1      60659631.0   \n",
       "23652              6512  NaN  ENSG00000231252          1      60659631.0   \n",
       "...                 ...  ...              ...        ...             ...   \n",
       "360156             1158  NaN  ENSG00000293035         21       9781848.0   \n",
       "360346                0  NaN  ENSG00000220891         22      22559343.0   \n",
       "360409             8382  NaN  ENSG00000280424         22      46006616.0   \n",
       "360410             5237  NaN  ENSG00000280424         22      46006616.0   \n",
       "360411                0  NaN  ENSG00000280424         22      46006616.0   \n",
       "\n",
       "        end_position strand  \n",
       "23604     63119542.0      -  \n",
       "23614      7450544.0      +  \n",
       "23650     60868041.0      -  \n",
       "23651     60868041.0      -  \n",
       "23652     60868041.0      -  \n",
       "...              ...    ...  \n",
       "360156     9839362.0      -  \n",
       "360346    22566602.0      +  \n",
       "360409    46010777.0      +  \n",
       "360410    46010777.0      +  \n",
       "360411    46010777.0      +  \n",
       "\n",
       "[5084 rows x 13 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cross3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5686434\n",
      "5608716\n",
      "5608716\n"
     ]
    }
   ],
   "source": [
    "final_merged_tfs = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cre_tf.csv')\n",
    "print(len(final_merged_tfs))\n",
    "final_merged_tfs = final_merged_tfs.drop_duplicates(subset=['chr', 'start', 'end', 'transcription_factor', 'cell_type', 'condition'])\n",
    "print(len(final_merged_tfs))\n",
    "final_merged_tfs = final_merged_tfs[final_merged_tfs[['chr','start', 'end','transcription_factor', 'cell_type', 'condition']].notna().any(axis=1)]\n",
    "print(len(final_merged_tfs))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, start, end, transcription_factor, cell_type, condition]\n",
       "Index: []"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cross4 = pd.merge(final_merged_tfs, final_merged, on=['chr', 'start', 'end'], how='left')\n",
    "cross4 = cross4[cross4['chr'].isna()]\n",
    "cross4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_merged_tfs.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cre_tf.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "114211\n",
      "83923\n",
      "83923\n",
      "138\n"
     ]
    }
   ],
   "source": [
    "final_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/pathways.csv')\n",
    "final_gene_pathways = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/combined_gene_pathways.csv')\n",
    "print(len(final_gene_pathways))\n",
    "final_gene_pathways = final_gene_pathways.drop_duplicates(subset=['entrez', 'pathway'])\n",
    "print(len(final_gene_pathways))\n",
    "final_gene_pathways = final_gene_pathways[final_gene_pathways[['entrez','pathway']].notna().any(axis=1)]\n",
    "print(len(final_gene_pathways))\n",
    "cross5 = pd.merge(final_gene_pathways, final_genes, on=['entrez'], how='left')\n",
    "cols = ['hgnc', 'ensembl_gene_id', 'chromosome', 'start_position', 'end_position', 'strand']\n",
    "cross5 = cross5[cross5[cols].isna().all(axis=1)]\n",
    "print(len(cross5))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3669</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>91543</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3433</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3429</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3434</td>\n",
       "      <td>REACTOME INTERFERON ALPHA BETA SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114198</th>\n",
       "      <td>6932</td>\n",
       "      <td>WNT SIGNALING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114200</th>\n",
       "      <td>2066</td>\n",
       "      <td>SIG PIP3 SIGNALING IN CARDIAC MYOCTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114201</th>\n",
       "      <td>57144</td>\n",
       "      <td>SIG PIP3 SIGNALING IN CARDIAC MYOCTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114206</th>\n",
       "      <td>5336</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114207</th>\n",
       "      <td>7409</td>\n",
       "      <td>SIG PIP3 SIGNALING IN B LYMPHOCYTES</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>83785 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        entrez                                   pathway\n",
       "0         3669  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "1        91543  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "2         3433  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "3         3429  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "4         3434  REACTOME INTERFERON ALPHA BETA SIGNALING\n",
       "...        ...                                       ...\n",
       "114198    6932                             WNT SIGNALING\n",
       "114200    2066     SIG PIP3 SIGNALING IN CARDIAC MYOCTES\n",
       "114201   57144     SIG PIP3 SIGNALING IN CARDIAC MYOCTES\n",
       "114206    5336       SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "114207    7409       SIG PIP3 SIGNALING IN B LYMPHOCYTES\n",
       "\n",
       "[83785 rows x 2 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "excluded_entrez_ids = cross5['entrez'].dropna().unique()\n",
    "excluded_entrez_ids\n",
    "filtered_gene_pathways = final_gene_pathways[~final_gene_pathways['entrez'].isin(excluded_entrez_ids)]\n",
    "filtered_gene_pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_gene_pathways.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/filtered_gene_pathways.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "525\n",
      "519\n"
     ]
    }
   ],
   "source": [
    "final_tfs = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/tfs.csv')\n",
    "print(len(final_tfs))\n",
    "final_tfs = final_tfs.drop_duplicates(subset=['transcription_factor'])\n",
    "print(len(final_tfs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>condition</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1397971</td>\n",
       "      <td>1400475</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1614037</td>\n",
       "      <td>1616691</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1745525</td>\n",
       "      <td>1747386</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2226872</td>\n",
       "      <td>2228282</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>ESC</td>\n",
       "      <td>IFN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608711</th>\n",
       "      <td>Y</td>\n",
       "      <td>13751202</td>\n",
       "      <td>13752042</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608712</th>\n",
       "      <td>Y</td>\n",
       "      <td>14523314</td>\n",
       "      <td>14525373</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608713</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840305</td>\n",
       "      <td>14840848</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608714</th>\n",
       "      <td>Y</td>\n",
       "      <td>19075637</td>\n",
       "      <td>19076488</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608715</th>\n",
       "      <td>Y</td>\n",
       "      <td>19566739</td>\n",
       "      <td>19568170</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>coculture</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5608716 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr     start       end transcription_factor cell_type  condition\n",
       "0         1    826435    828307              AHRARNT       ESC        IFN\n",
       "1         1   1397971   1400475              AHRARNT       ESC        IFN\n",
       "2         1   1614037   1616691              AHRARNT       ESC        IFN\n",
       "3         1   1745525   1747386              AHRARNT       ESC        IFN\n",
       "4         1   2226872   2228282              AHRARNT       ESC        IFN\n",
       "...      ..       ...       ...                  ...       ...        ...\n",
       "5608711   Y  13751202  13752042               ZFP961      iPSC  coculture\n",
       "5608712   Y  14523314  14525373               ZFP961      iPSC  coculture\n",
       "5608713   Y  14840305  14840848               ZFP961      iPSC  coculture\n",
       "5608714   Y  19075637  19076488               ZFP961      iPSC  coculture\n",
       "5608715   Y  19566739  19568170               ZFP961      iPSC  coculture\n",
       "\n",
       "[5608716 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m1 = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cre_tf.csv')\n",
    "m1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>826435</td>\n",
       "      <td>828307</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1397971</td>\n",
       "      <td>1400475</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1614037</td>\n",
       "      <td>1616691</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1745525</td>\n",
       "      <td>1747386</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2226872</td>\n",
       "      <td>2228282</td>\n",
       "      <td>AHRARNT</td>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608711</th>\n",
       "      <td>Y</td>\n",
       "      <td>13751202</td>\n",
       "      <td>13752042</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608712</th>\n",
       "      <td>Y</td>\n",
       "      <td>14523314</td>\n",
       "      <td>14525373</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608713</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840305</td>\n",
       "      <td>14840848</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608714</th>\n",
       "      <td>Y</td>\n",
       "      <td>19075637</td>\n",
       "      <td>19076488</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5608715</th>\n",
       "      <td>Y</td>\n",
       "      <td>19566739</td>\n",
       "      <td>19568170</td>\n",
       "      <td>ZFP961</td>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5608716 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr     start       end transcription_factor  condition cell_type\n",
       "0         1    826435    828307              AHRARNT        IFN       ESC\n",
       "1         1   1397971   1400475              AHRARNT        IFN       ESC\n",
       "2         1   1614037   1616691              AHRARNT        IFN       ESC\n",
       "3         1   1745525   1747386              AHRARNT        IFN       ESC\n",
       "4         1   2226872   2228282              AHRARNT        IFN       ESC\n",
       "...      ..       ...       ...                  ...        ...       ...\n",
       "5608711   Y  13751202  13752042               ZFP961  coculture      iPSC\n",
       "5608712   Y  14523314  14525373               ZFP961  coculture      iPSC\n",
       "5608713   Y  14840305  14840848               ZFP961  coculture      iPSC\n",
       "5608714   Y  19075637  19076488               ZFP961  coculture      iPSC\n",
       "5608715   Y  19566739  19568170               ZFP961  coculture      iPSC\n",
       "\n",
       "[5608716 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m1 = m1[['chr', 'start', 'end', 'transcription_factor','condition','cell_type']]\n",
    "m1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "m1.to_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/merged_cre_tf.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Post Checks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "u1 = pd.read_csv('/Users/nathan/Desktop/Bioinformatics/Projects/AD_Database/Tables/unique_cres_genes.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entrez</th>\n",
       "      <th>distance_to_tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IFN</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>89295608</td>\n",
       "      <td>89296790</td>\n",
       "      <td>3433</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41773</th>\n",
       "      <td>IFN</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>7</td>\n",
       "      <td>107907754</td>\n",
       "      <td>107908346</td>\n",
       "      <td>1738</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97596</th>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>86714950</td>\n",
       "      <td>86715643</td>\n",
       "      <td>11155</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100223</th>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>20</td>\n",
       "      <td>9397627</td>\n",
       "      <td>9398557</td>\n",
       "      <td>5332</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101325</th>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>15434533</td>\n",
       "      <td>15435355</td>\n",
       "      <td>11330</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102516</th>\n",
       "      <td>TREM2KO</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>119641448</td>\n",
       "      <td>119642807</td>\n",
       "      <td>200373</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142021</th>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>19</td>\n",
       "      <td>49331583</td>\n",
       "      <td>49332084</td>\n",
       "      <td>951</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150772</th>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>10</td>\n",
       "      <td>86714950</td>\n",
       "      <td>86715643</td>\n",
       "      <td>11155</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153429</th>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>20</td>\n",
       "      <td>9397627</td>\n",
       "      <td>9398557</td>\n",
       "      <td>5332</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154525</th>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>15434533</td>\n",
       "      <td>15435355</td>\n",
       "      <td>11330</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155663</th>\n",
       "      <td>TREM2R47H</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>119641448</td>\n",
       "      <td>119642807</td>\n",
       "      <td>200373</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170818</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>21</td>\n",
       "      <td>42539890</td>\n",
       "      <td>42540430</td>\n",
       "      <td>54020</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176914</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3</td>\n",
       "      <td>38349901</td>\n",
       "      <td>38350375</td>\n",
       "      <td>9942</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180601</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>1</td>\n",
       "      <td>33085992</td>\n",
       "      <td>33086852</td>\n",
       "      <td>113451</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188302</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5</td>\n",
       "      <td>131536119</td>\n",
       "      <td>131537013</td>\n",
       "      <td>51735</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194883</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>4</td>\n",
       "      <td>127629670</td>\n",
       "      <td>127629870</td>\n",
       "      <td>27152</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213575</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>2</td>\n",
       "      <td>11175098</td>\n",
       "      <td>11175987</td>\n",
       "      <td>130814</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214263</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>19</td>\n",
       "      <td>45325961</td>\n",
       "      <td>45326707</td>\n",
       "      <td>1158</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220055</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>1</td>\n",
       "      <td>149967349</td>\n",
       "      <td>149968174</td>\n",
       "      <td>56957</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250087</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3</td>\n",
       "      <td>128572467</td>\n",
       "      <td>128572999</td>\n",
       "      <td>23434</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257677</th>\n",
       "      <td>xenot7d</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>13</td>\n",
       "      <td>20099338</td>\n",
       "      <td>20099614</td>\n",
       "      <td>104355137</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271378</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>21</td>\n",
       "      <td>42539890</td>\n",
       "      <td>42540430</td>\n",
       "      <td>54020</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277353</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3</td>\n",
       "      <td>38349901</td>\n",
       "      <td>38350375</td>\n",
       "      <td>9942</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280984</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>1</td>\n",
       "      <td>33085992</td>\n",
       "      <td>33086852</td>\n",
       "      <td>113451</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288538</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>5</td>\n",
       "      <td>131536119</td>\n",
       "      <td>131537013</td>\n",
       "      <td>51735</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294975</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>4</td>\n",
       "      <td>127629670</td>\n",
       "      <td>127629870</td>\n",
       "      <td>27152</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313308</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>2</td>\n",
       "      <td>11175098</td>\n",
       "      <td>11175987</td>\n",
       "      <td>130814</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313986</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>19</td>\n",
       "      <td>45325961</td>\n",
       "      <td>45326707</td>\n",
       "      <td>1158</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>319655</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>1</td>\n",
       "      <td>149967349</td>\n",
       "      <td>149968174</td>\n",
       "      <td>56957</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349209</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>3</td>\n",
       "      <td>128572467</td>\n",
       "      <td>128572999</td>\n",
       "      <td>23434</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356732</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>13</td>\n",
       "      <td>20099338</td>\n",
       "      <td>20099614</td>\n",
       "      <td>104355137</td>\n",
       "      <td>3087</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr      start        end     entrez  \\\n",
       "0             IFN       ESC  10   89295608   89296790       3433   \n",
       "41773         IFN      iPSC   7  107907754  107908346       1738   \n",
       "97596     TREM2KO       ESC  10   86714950   86715643      11155   \n",
       "100223    TREM2KO       ESC  20    9397627    9398557       5332   \n",
       "101325    TREM2KO       ESC   1   15434533   15435355      11330   \n",
       "102516    TREM2KO       ESC   2  119641448  119642807     200373   \n",
       "142021  TREM2R47H       ESC  19   49331583   49332084        951   \n",
       "150772  TREM2R47H       ESC  10   86714950   86715643      11155   \n",
       "153429  TREM2R47H       ESC  20    9397627    9398557       5332   \n",
       "154525  TREM2R47H       ESC   1   15434533   15435355      11330   \n",
       "155663  TREM2R47H       ESC   2  119641448  119642807     200373   \n",
       "170818    xenot7d      iPSC  21   42539890   42540430      54020   \n",
       "176914    xenot7d      iPSC   3   38349901   38350375       9942   \n",
       "180601    xenot7d      iPSC   1   33085992   33086852     113451   \n",
       "188302    xenot7d      iPSC   5  131536119  131537013      51735   \n",
       "194883    xenot7d      iPSC   4  127629670  127629870      27152   \n",
       "213575    xenot7d      iPSC   2   11175098   11175987     130814   \n",
       "214263    xenot7d      iPSC  19   45325961   45326707       1158   \n",
       "220055    xenot7d      iPSC   1  149967349  149968174      56957   \n",
       "250087    xenot7d      iPSC   3  128572467  128572999      23434   \n",
       "257677    xenot7d      iPSC  13   20099338   20099614  104355137   \n",
       "271378  coculture      iPSC  21   42539890   42540430      54020   \n",
       "277353  coculture      iPSC   3   38349901   38350375       9942   \n",
       "280984  coculture      iPSC   1   33085992   33086852     113451   \n",
       "288538  coculture      iPSC   5  131536119  131537013      51735   \n",
       "294975  coculture      iPSC   4  127629670  127629870      27152   \n",
       "313308  coculture      iPSC   2   11175098   11175987     130814   \n",
       "313986  coculture      iPSC  19   45325961   45326707       1158   \n",
       "319655  coculture      iPSC   1  149967349  149968174      56957   \n",
       "349209  coculture      iPSC   3  128572467  128572999      23434   \n",
       "356732  coculture      iPSC  13   20099338   20099614  104355137   \n",
       "\n",
       "        distance_to_tss  \n",
       "0                  3087  \n",
       "41773              3087  \n",
       "97596              3087  \n",
       "100223             3087  \n",
       "101325             3087  \n",
       "102516             3087  \n",
       "142021             3087  \n",
       "150772             3087  \n",
       "153429             3087  \n",
       "154525             3087  \n",
       "155663             3087  \n",
       "170818             3087  \n",
       "176914             3087  \n",
       "180601             3087  \n",
       "188302             3087  \n",
       "194883             3087  \n",
       "213575             3087  \n",
       "214263             3087  \n",
       "220055             3087  \n",
       "250087             3087  \n",
       "257677             3087  \n",
       "271378             3087  \n",
       "277353             3087  \n",
       "280984             3087  \n",
       "288538             3087  \n",
       "294975             3087  \n",
       "313308             3087  \n",
       "313986             3087  \n",
       "319655             3087  \n",
       "349209             3087  \n",
       "356732             3087  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "u1_3087 = u1[u1['distance_to_tss'] == 3087]\n",
    "u1_3087 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>condition</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entrez</th>\n",
       "      <th>distance_to_tss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>261570</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>10</td>\n",
       "      <td>89282690</td>\n",
       "      <td>89283817</td>\n",
       "      <td>3433</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261571</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>10</td>\n",
       "      <td>89295629</td>\n",
       "      <td>89296902</td>\n",
       "      <td>3433</td>\n",
       "      <td>3108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261572</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>10</td>\n",
       "      <td>89301191</td>\n",
       "      <td>89302637</td>\n",
       "      <td>3433</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261573</th>\n",
       "      <td>coculture</td>\n",
       "      <td>iPSC</td>\n",
       "      <td>10</td>\n",
       "      <td>89302738</td>\n",
       "      <td>89303106</td>\n",
       "      <td>3433</td>\n",
       "      <td>693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        condition cell_type chr     start       end  entrez  distance_to_tss\n",
       "261570  coculture      iPSC  10  89282690  89283817    3433                0\n",
       "261571  coculture      iPSC  10  89295629  89296902    3433             3108\n",
       "261572  coculture      iPSC  10  89301191  89302637    3433                0\n",
       "261573  coculture      iPSC  10  89302738  89303106    3433              693"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "u2 = u1[(u1['entrez'] == 3433) & (u1['condition'] == 'coculture')]\n",
    "u2"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
